{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# to rerun any part of this code\n",
    "import os\n",
    "import json\n",
    "import pickle\n",
    "import time\n",
    "import gzip\n",
    "import shutil\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def generate_gwas_output_as_tsv_file(inVCF_File, writeFile=None):\n",
    "    root = '../tabular_format_gwas_data/'\n",
    "    t0 = time.time()\n",
    "    if writeFile is None:\n",
    "        writeFile = root + inVCF_File.split('/')[-1].split('.')[0] + '.tsv'\n",
    "    \n",
    "    print(\"Processing:\", inVCF_File)\n",
    "    if \".gz\" == inVCF_File[-3:]:\n",
    "        with gzip.open(inVCF_File, 'rb') as f_in:\n",
    "            with open(root+'temp_vcf.vcf', 'wb') as f_out:\n",
    "                shutil.copyfileobj(f_in, f_out)\n",
    "        inVCF_File = root+'temp_vcf.vcf'\n",
    "    print(\"gzip open done :\", inVCF_File)\n",
    "    \"\"\"\n",
    "    Convert all the VCF into simplified tsv format\n",
    "    Source: https://github.com/everestial/VCF-simplify#table-of-contents\n",
    "    python3 ./VCF-Simplify/VCF-Simplify/VcfSimplify.py SimplifyVCF -toType table -inVCF ./ukb-b-6134.vcf -outFile ./simple_table.txt\n",
    "    \"\"\"\n",
    "    vcf_simplify_path = \"./VCF-Simplify/VCF-Simplify/VcfSimplify.py\"\n",
    "    out_File = root+'temp_table.tsv'\n",
    "    os.system(f\"python3 {vcf_simplify_path} SimplifyVCF -toType table -inVCF {inVCF_File} -outFile {out_File}\")\n",
    "    print(writeFile)\n",
    "        \n",
    "    f_read = open(out_File, 'r')\n",
    "    f_write = open(writeFile, 'w')\n",
    "    cnt = 0\n",
    "    for line in f_read:\n",
    "#         print(line)\n",
    "        newliner = ''\n",
    "        if '\\n' == line[-1]:\n",
    "            newliner = '\\n'\n",
    "            line = line.strip()\n",
    "        if 'CHROM' in line:\n",
    "            line = line.upper() + '\\tPVAL_generated_from_LP'\n",
    "        else:\n",
    "            LP = float(line.split('\\t')[10])\n",
    "#             LP = float(line.split('\\t')[14])\n",
    "            #find out LP column from the tsv file\n",
    "            \n",
    "            p_val = str(10**(-1 * LP))\n",
    "            line = line + '\\t' + p_val\n",
    "        line = line + newliner\n",
    "    #     print(line)\n",
    "        f_write.write(line)\n",
    "        cnt += 1\n",
    "    #     if cnt == 20: break\n",
    "\n",
    "    f_read.close()\n",
    "    f_write.close()\n",
    "    print(f'Total SNPs: {cnt} | Total exec time: {time.time() - t0} seconds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_excel('../GWAS_datasets_to_consider.xlsx', engine='openpyxl')[['GWAS_ID', 'Trait', 'Category']]\n",
    "ALL_GWAS_IDS = list(df['GWAS_ID'].to_numpy().tolist()) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing: ./tabular_format_gwas_data/ieu-b-5067.vcf.gz\n",
      "gzip open done : ./tabular_format_gwas_data/temp_vcf.vcf\n",
      "    ## VCF Simplify ## : Python application for parsing VCF files.\n",
      "    Author: Bishwa K. Giri\n",
      "    Contributors: Bhuwan Aryal\n",
      "    Contact: bkgiri@uncg.edu, kirannbishwa01@gmail.com, bhuwanaryal19@gmail.com\n",
      "\n",
      "Using the following arguments: \n",
      "Namespace(GTbase=['GT:numeric'], PG='PG', PI='PI', formats=['all'], inVCF='./tabular_format_gwas_data/temp_vcf.vcf', includeUnphased='0', infos=['all'], mode='0', outFile='./tabular_format_gwas_data/temp_table.tsv', outHeaderName=None, preHeader=['all'], samples='all', toType=['table'])\n",
      "\n",
      "Using option \"SimplifyVCF\"\n",
      "  Simplifying the VCF records ...\n",
      "1 samples found\n",
      "\n",
      "Skipping the header.\n",
      "\n",
      "Writing table in a wide format.\n",
      "\n",
      "sample genotypes tag 'GT' are written as 'numeric' bases\n",
      "\n",
      "- Output header of the simplified VCF file:\n",
      "['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO:AF', 'ieu-b-5067:ES', 'ieu-b-5067:SE', 'ieu-b-5067:LP', 'ieu-b-5067:AF', 'ieu-b-5067:SS', 'ieu-b-5067:EZ', 'ieu-b-5067:SI', 'ieu-b-5067:NC', 'ieu-b-5067:ID']\n",
      "\n",
      "Reading the input vcf file ... \n",
      "\n",
      "Contig 1 is being processed ... \n",
      "\n",
      "Contig 2 is being processed ... \n",
      "\n",
      "Contig 3 is being processed ... \n",
      "\n",
      "Contig 4 is being processed ... \n",
      "\n",
      "Contig 5 is being processed ... \n",
      "\n",
      "Contig 6 is being processed ... \n",
      "\n",
      "Contig 7 is being processed ... \n",
      "\n",
      "Contig 8 is being processed ... \n",
      "\n",
      "Contig 9 is being processed ... \n",
      "\n",
      "Contig 10 is being processed ... \n",
      "\n",
      "Contig 11 is being processed ... \n",
      "\n",
      "Contig 12 is being processed ... \n",
      "\n",
      "Contig 13 is being processed ... \n",
      "\n",
      "Contig 14 is being processed ... \n",
      "\n",
      "Contig 15 is being processed ... \n",
      "\n",
      "Contig 16 is being processed ... \n",
      "\n",
      "Contig 17 is being processed ... \n",
      "\n",
      "Contig 18 is being processed ... \n",
      "\n",
      "Contig 19 is being processed ... \n",
      "\n",
      "Contig 20 is being processed ... \n",
      "\n",
      "Contig 21 is being processed ... \n",
      "\n",
      "Contig 22 is being processed ... \n",
      "\n",
      "Contig X is being processed ... \n",
      "\n",
      "elapsed time:  103.18159770965576\n",
      "Completed converting the VCF file to table output.\n",
      "fnc_vcf_to_table: memory before: 10,380, after: 10,692, consumed: 312; exec time: 103.18516254425049 seconds\n",
      "./tabular_format_gwas_data/ieu-b-5067.tsv\n",
      "Total SNPs: 12321876 | Total exec time: 134.22045969963074 seconds\n"
     ]
    }
   ],
   "source": [
    "# ukb-b-4171\n",
    "# ukb-b-4000\n",
    "# ukb-d-20544_1\n",
    "# ukb-d-20544_15\n",
    "# ukb-d-20441\n",
    "# ukb-b-18162\n",
    "# ukb-b-14624\n",
    "# ukb-b-9667\n",
    "\n",
    "\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-4171.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ieu-b-5067.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-18162.vcf.gz', writeFile=None)\n",
    "\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-14624.vcf.gz', writeFile=None)\n",
    "\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-9667.vcf.gz', writeFile=None)\n",
    "\n",
    "\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-d-20544_1.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-d-20544_15.vcf.gz', writeFile=None)\n",
    "\n",
    "\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-12417.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-11495.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-12493.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-19732.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-17006.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-5779.vcf.gz', writeFile=None)\n",
    "# generate_gwas_output_as_tsv_file(inVCF_File='./tabular_format_gwas_data/ukb-b-15541.vcf.gz', writeFile=None)\n",
    "# need to just specify the file path here\n",
    "# ebi-a-GCST005920.vcf.gz\n",
    "# ebi-a-GCST005923.vcf.gz\n",
    "# ukb-b-14699.vcf.gz\n",
    "# ukb-b-323.vcf.gz\n",
    "# ukb-b-6358\n",
    "# next update in the 'GWAS_datasets_to_consider.xlsx'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# %pip install openpyxl\n",
    "import pandas as pd\n",
    "\n",
    "category_map = {\n",
    "    'Continuous' : 'beta', \n",
    "    'Categorical Ordered (assumed continuous)': 'beta',\n",
    "    'Binary': 'or',\n",
    "    'NA (Possibly binary)': 'or'\n",
    "}\n",
    "\n",
    "# always replace Binary traits category with 'or' and continuous category with 'beta'\n",
    "\n",
    "df = pd.read_excel('../GWAS_datasets_to_consider.xlsx', engine='openpyxl')[['GWAS_ID', 'Trait', 'Category']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "53\n"
     ]
    }
   ],
   "source": [
    "from glob import glob\n",
    "# df = pd.read_excel('GWAS_datasets_to_consider.xlsx', engine='openpyxl')[['GWAS_ID', 'Trait', 'Category']]\n",
    "# all_gwas_vcf_files = glob('/mnt/c/Users/HP/Downloads/Compressed/GWAS/*')\n",
    "# ROOT = '/mnt/c/Users/HP/Downloads/Compressed/GWAS/'\n",
    "# ALL_GWAS_IDS = list(set(df['GWAS_ID'].to_numpy().tolist()) - {'bbj-a-46', 'bbj-a-78', 'ieu-a-1283', 'ukb-a-257','ukb-b-2209','ukb-b-17627','ieub-109','ieu-b-110','ieu-b-111','met-d-Total_C'})\n",
    "# bbj-a-78,ukb-a-257,bbj-a-46\n",
    "# ALL_GWAS_IDS = list(set(df['GWAS_ID'].to_numpy().tolist()) - { 'ieu-a-1283','bbj-a-46', 'bbj-a-78'})\n",
    "# ALL_GWAS_IDS = list(set(df['GWAS_ID'].to_numpy().tolist()) - {  'ieu-b-109','ieu-b-110','ieu-b-111','met-d-Total_C','ukb-b-2209','ukb-b-17627','ukb-a-257','ieu-a-1283','bbj-a-46', 'bbj-a-78','ukb-b-14180','ukb-b-13806','ukb-b-12064','ukb-b-323','ukb-b-14699','ukb-b-14699','ukb-b-5779'})\n",
    "ALL_GWAS_IDS = list(df['GWAS_ID'].to_numpy().tolist()) \n",
    "\n",
    "# print(len(ALL_GWAS_IDS))\n",
    "# print(ALL_GWAS_IDS)\n",
    "\n",
    "base_files = {}\n",
    "# take all except for AD PRS\n",
    "\n",
    "for GWAS_ID in ALL_GWAS_IDS:\n",
    "#     if GWAS_ID=='ieu-b-5067':\n",
    "#         continue\n",
    "#     file = ROOT + GWAS_ID + '.vcf.gz'\n",
    "#     base_files[GWAS_ID] = category_map[df[df['GWAS_ID'] == GWAS_ID].to_numpy()[0, 2]]\n",
    "    base_files[GWAS_ID] = df[df['GWAS_ID'] == GWAS_ID].to_numpy()[0, 2]\n",
    "#     base_files[GWAS_ID]=category_map[base_files[GWAS_ID]]\n",
    "    \n",
    "print(base_files.__len__())\n",
    "#     generate_gwas_output_as_tsv_file(inVCF_File=file, writeFile=None)\n",
    "# print(name_mapping)\n",
    "# print(name_mapping)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'ukb-b-13806': 'or', 'ukb-d-20405_0': 'or', 'ukb-d-20405_1': 'or', 'ukb-d-20405_2': 'or', 'met-d-Total_C': 'beta', 'ieu-b-109': 'beta', 'ieu-b-110': 'beta', 'ieu-b-111': 'beta', 'ieu-b-25': 'beta', 'ieu-b-38': 'beta', 'ieu-b-39': 'beta', 'ukb-a-257': 'or', 'ukb-b-12064': 'or', 'ukb-b-17627': 'beta', 'ukb-b-18275': 'or', 'ukb-b-19953': 'beta', 'ukb-b-2209': 'beta', 'ukb-b-3957': 'beta', 'ukb-b-4424': 'beta', 'ukb-b-6134': 'beta', 'ukb-b-6324': 'beta', 'ukb-b-7663': 'or', 'ukb-b-770': 'beta', 'ukb-b-8476': 'or', 'ukb-b-323': 'or', 'ukb-b-14699': 'or', 'ukb-b-14180': 'or', 'ukb-b-17243': 'or', 'ukb-b-6358': 'or', 'ukb-b-17006': 'or', 'ukb-b-5779': 'beta', 'ukb-b-15541': 'or', 'ukb-b-19732': 'or', 'ukb-b-20289': 'or', 'ukb-b-14057': 'or', 'ukb-b-12963': 'beta', 'ukb-b-12417': 'beta', 'ukb-b-11495': 'or', 'ukb-b-12493': 'or', 'ukb-b-4282': 'beta', 'ukb-b-4171': 'or', 'ukb-b-4000': 'or', 'ukb-d-20544_1': 'or', 'ukb-d-20544_15': 'or', 'ukb-b-20556': 'or', 'ukb-d-20441': 'or', 'ukb-b-18162': 'or', 'ukb-b-14624': 'or', 'ukb-b-9667': 'or', 'ieu-a-1183': 'or', 'ukb-d-30500_irnt': 'beta', 'ieu-b-7': 'or', 'ieu-b-5067': 'beta'}\n",
      "ukb-b-13806\n",
      "ukb-d-20405_0\n",
      "ukb-d-20405_1\n",
      "ukb-d-20405_2\n",
      "met-d-Total_C\n",
      "ieu-b-109\n",
      "ieu-b-110\n",
      "ieu-b-111\n",
      "ieu-b-25\n",
      "ieu-b-38\n",
      "ieu-b-39\n",
      "ukb-a-257\n",
      "ukb-b-12064\n",
      "ukb-b-17627\n",
      "ukb-b-18275\n",
      "ukb-b-19953\n",
      "ukb-b-2209\n",
      "ukb-b-3957\n",
      "ukb-b-4424\n",
      "ukb-b-6134\n",
      "ukb-b-6324\n",
      "ukb-b-7663\n",
      "ukb-b-770\n",
      "ukb-b-8476\n",
      "ukb-b-323\n",
      "ukb-b-14699\n",
      "ukb-b-14180\n",
      "ukb-b-17243\n",
      "ukb-b-6358\n",
      "ukb-b-17006\n",
      "ukb-b-5779\n",
      "ukb-b-15541\n",
      "ukb-b-19732\n",
      "ukb-b-20289\n",
      "ukb-b-14057\n",
      "ukb-b-12963\n",
      "ukb-b-12417\n",
      "ukb-b-11495\n",
      "ukb-b-12493\n",
      "ukb-b-4282\n",
      "ukb-b-4171\n",
      "ukb-b-4000\n",
      "ukb-d-20544_1\n",
      "ukb-d-20544_15\n",
      "ukb-b-20556\n",
      "ukb-d-20441\n",
      "ukb-b-18162\n",
      "ukb-b-14624\n",
      "ukb-b-9667\n",
      "ieu-a-1183\n",
      "ukb-d-30500_irnt\n",
      "ieu-b-7\n",
      "ieu-b-5067\n"
     ]
    }
   ],
   "source": [
    "# base_files\n",
    "print(base_files)\n",
    "for GWAS_ID in base_files:\n",
    "    print(GWAS_ID)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    }
   ],
   "source": [
    "# gender_map = {'Female': 0,'Male': 1}\n",
    "# ADNIMERGE = pd.read_csv('ADNIMERGE.csv')\n",
    "\n",
    "\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx')\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "\n",
    "def get_gender_and_age(projid):\n",
    "    gender = basic_rosmap[basic_rosmap['projid'] == projid]['msex'].dropna().unique()[0]\n",
    "    age = basic_rosmap[basic_rosmap['projid'] == projid]['msex'].dropna().unique()[0]\n",
    "    return str(gender) + ' ' + str(age)\n",
    "\n",
    "# rosmap_total_non_genetic=pd.merge(basic_rosmap,long_rosmap,on='projid')\n",
    "\n",
    "# print(rosmap_total_non_genetic)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# For ROSMAP Affymetrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def run_prsice(GWAS_ID, redo=False):\n",
    "    TARGET_DATA = 'rosmap_affymetrix' # new and larger dataset\n",
    "    PRSice_output_ROSMAP=\"./PRSice_output_ROSMAP\"\n",
    "    prsice_linux=\"../PRSice_linux\"\n",
    "    '''\n",
    "    return_status --> 0: success | 256: Error. Execusion Halted | 2: \n",
    "    '''\n",
    "#     need to alter between these two if needed\n",
    "    if not os.path.isdir(f'./PRSice_output_ROSMAP/{GWAS_ID}/'):\n",
    "#     if True:\n",
    "        print(\"creating directory\")\n",
    "        return_status = os.system(f'mkdir ./PRSice_output_ROSMAP/{GWAS_ID}/')\n",
    "        print('0:', return_status)\n",
    "        if return_status == 256 and not redo:\n",
    "            return return_status\n",
    "    print('stat',{base_files[GWAS_ID]})\n",
    "\n",
    "    if os.path.isfile(f'{PRSice_output_ROSMAP}/{GWAS_ID}/{GWAS_ID}.valid'):\n",
    "        prsice_command = f'Rscript \"{prsice_linux}/PRSice.R\" \\\n",
    "            --dir {PRSice_output_ROSMAP} \\\n",
    "            --prsice \"{prsice_linux}/PRSice_linux\" \\\n",
    "            --base ../tabular_format_gwas_data/{GWAS_ID}.tsv \\\n",
    "            --target {TARGET_DATA} \\\n",
    "            --thread 2 \\\n",
    "            --stat {base_files[GWAS_ID].upper()} \\\n",
    "            --{base_files[GWAS_ID]} \\\n",
    "            --binary-target F \\\n",
    "            --quantile 10 \\\n",
    "            --out {PRSice_output_ROSMAP}/{GWAS_ID}/{GWAS_ID} \\\n",
    "            --snp ID --chr CHROM --bp POS --A2 REF --A1 ALT --stat {GWAS_ID.upper()}:ES --pvalue PVAL_generated_from_LP \\\n",
    "            --score std \\\n",
    "            --cov covar_affymetrix.txt \\\n",
    "            --extract \"{PRSice_output_ROSMAP}/{GWAS_ID}/{GWAS_ID}.valid\"'\n",
    "    else :\n",
    "        prsice_command = f'Rscript \"{prsice_linux}/PRSice.R\"\\\n",
    "            --dir {PRSice_output_ROSMAP} \\\n",
    "            --prsice \"{prsice_linux}/PRSice_linux\" \\\n",
    "            --base ../tabular_format_gwas_data/{GWAS_ID}.tsv \\\n",
    "            --target {TARGET_DATA} \\\n",
    "            --thread 2 \\\n",
    "            --stat {base_files[GWAS_ID].upper()} \\\n",
    "            --{base_files[GWAS_ID]} \\\n",
    "            --binary-target F \\\n",
    "            --quantile 10 \\\n",
    "            --out {PRSice_output_ROSMAP}/{GWAS_ID}/{GWAS_ID} \\\n",
    "            --snp ID --chr CHROM --bp POS --A2 REF --A1 ALT --stat {GWAS_ID.upper()}:ES --pvalue PVAL_generated_from_LP \\\n",
    "            --score std \\\n",
    "            --cov covar_affymetrix.txt '\n",
    "#     # in later steps, select the PRS values for the selected p-value-threshold for all the samples (both training and testing)\n",
    "#     print(prsice_command)\n",
    "    print('stat',{base_files[GWAS_ID]})\n",
    "    return_status = os.system(prsice_command) # generate the file named {GWAS_ID}.valid\n",
    "    print('1:', return_status)\n",
    "    return_status = os.system('echo \"===== Done =====\"')\n",
    "    print('2:', return_status) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_prsice_new(GWAS_ID, redo=False):\n",
    "    TARGET_DATA = 'rosmap_affymetrix' # new and larger dataset\n",
    "    PRSice_output_ROSMAP=\"./PRSice_output_ROSMAP_2\"\n",
    "    prsice_linux=\"../PRSice_linux\"\n",
    "    '''\n",
    "    return_status --> 0: success | 256: Error. Execusion Halted | 2: \n",
    "    '''\n",
    "#     need to alter between these two if needed\n",
    "    if not os.path.isdir(f'./PRSice_output_ROSMAP_2/{GWAS_ID}/'):\n",
    "#     if True:\n",
    "        print(\"creating directory\")\n",
    "        return_status = os.system(f'mkdir ./PRSice_output_ROSMAP_2/{GWAS_ID}/')\n",
    "        print('0:', return_status)\n",
    "        if return_status == 256 and not redo:\n",
    "            return return_status\n",
    "    print('stat',{base_files[GWAS_ID]})\n",
    "\n",
    "    if os.path.isfile(f'{PRSice_output_ROSMAP}/{GWAS_ID}/{GWAS_ID}.valid'):\n",
    "        prsice_command = f'Rscript \"{prsice_linux}/PRSice.R\" \\\n",
    "            --dir {PRSice_output_ROSMAP} \\\n",
    "            --prsice \"{prsice_linux}/PRSice_linux\" \\\n",
    "            --base ../tabular_format_gwas_data/{GWAS_ID}.tsv \\\n",
    "            --target {TARGET_DATA} \\\n",
    "            --thread 2 \\\n",
    "            --stat {base_files[GWAS_ID].upper()} \\\n",
    "            --{base_files[GWAS_ID]} \\\n",
    "            --binary-target F \\\n",
    "            --quantile 10 \\\n",
    "            --out {PRSice_output_ROSMAP}/{GWAS_ID}/{GWAS_ID} \\\n",
    "            --snp ID --chr X --bp Y --A2 Z --A1 ALT --stat {GWAS_ID.upper()}:ES --pvalue PVAL_generated_from_LP \\\n",
    "            --score std \\\n",
    "            --cov covar_affymetrix.txt \\\n",
    "            --extract \"{PRSice_output_ROSMAP}/{GWAS_ID}/{GWAS_ID}.valid\"'\n",
    "    else :\n",
    "        prsice_command = f'Rscript \"{prsice_linux}/PRSice.R\"\\\n",
    "            --dir {PRSice_output_ROSMAP} \\\n",
    "            --prsice \"{prsice_linux}/PRSice_linux\" \\\n",
    "            --base ../tabular_format_gwas_data/{GWAS_ID}.tsv \\\n",
    "            --target {TARGET_DATA} \\\n",
    "            --thread 2 \\\n",
    "            --stat {base_files[GWAS_ID].upper()} \\\n",
    "            --{base_files[GWAS_ID]} \\\n",
    "            --binary-target F \\\n",
    "            --quantile 10 \\\n",
    "            --out {PRSice_output_ROSMAP}/{GWAS_ID}/{GWAS_ID} \\\n",
    "            --snp ID --chr X --bp Y --A2 Z --A1 ALT --stat {GWAS_ID.upper()}:ES --pvalue PVAL_generated_from_LP \\\n",
    "            --score std \\\n",
    "            --cov covar_affymetrix.txt '\n",
    "#     # in later steps, select the PRS values for the selected p-value-threshold for all the samples (both training and testing)\n",
    "#     print(prsice_command)\n",
    "    print('stat',{base_files[GWAS_ID]})\n",
    "    return_status = os.system(prsice_command) # generate the file named {GWAS_ID}.valid\n",
    "    print('1:', return_status)\n",
    "    return_status = os.system('echo \"===== Done =====\"')\n",
    "    print('2:', return_status) \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notes on run_prsice\n",
    "\n",
    "for each of the data ( individual ) that we have in target dataset, we use our built psrice framework to find out the prs value. so we now know the PRS for target dataset IDs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-11-07 14:08:10\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-a-257.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/temp/ukb-a-257/ukb-a-257 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2969503439 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-A-257:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-a-257 \n",
      "================================================== \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-a-257.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-A-257:ES\tUKB-A-257:SE\tUKB-A-257:LP\tUKB-A-257:AF\tUKB-A-257:SS\tUKB-A-257:EZ\tUKB-A-257:SI\tUKB-A-257:NC\tUKB-A-257:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 37.61%"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1: 2\n",
      "===== Done =====\n",
      "2: 0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "Reading 37.62%\r",
      "Reading 37.63%\r",
      "Reading 37.64%\r",
      "Reading 37.65%\r",
      "Reading 37.66%\r",
      "Reading 37.67%\r",
      "Reading 37.68%\r",
      "Reading 37.69%\r",
      "Reading 37.70%\r",
      "Reading 37.71%\r",
      "Reading 37.72%\r",
      "Reading 37.73%\r",
      "Reading 37.74%\r",
      "Reading 37.75%\r",
      "Reading 37.76%\r",
      "Reading 37.77%\r",
      "Reading 37.78%\r",
      "Reading 37.79%\r",
      "Reading 37.80%\r",
      "Reading 37.81%\r",
      "Reading 37.82%\r",
      "Reading 37.83%\r",
      "Reading 37.84%\r",
      "Reading 37.85%\r",
      "Reading 37.86%\r",
      "Reading 37.87%\r",
      "Reading 37.88%\r",
      "Reading 37.89%\r",
      "Reading 37.90%\r",
      "Reading 37.91%\r",
      "Reading 37.92%\r",
      "Reading 37.93%\r",
      "Reading 37.94%\r",
      "Reading 37.95%\r",
      "Reading 37.96%\r",
      "Reading 37.97%\r",
      "Reading 37.98%\r",
      "Reading 37.99%\r",
      "Reading 38.00%\r",
      "Reading 38.01%\r",
      "Reading 38.02%\r",
      "Reading 38.03%Error: \n",
      "Execution halted\n"
     ]
    }
   ],
   "source": [
    "run_prsice_new(GWAS_ID='ukb-a-257')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----------------------counter  1  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-13806\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:37:07\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-13806.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-13806/ukb-b-13806.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-13806/ukb-b-13806 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 1732038236 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-13806:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-13806 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-13806.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-13806:ES\tUKB-B-13806:SE\tUKB-B-13806:LP\tUKB-B-13806:AF\tUKB-B-13806:SS\tUKB-B-13806:EZ\tUKB-B-13806:SI\tUKB-B-13806:NC\tUKB-B-13806:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "4004523 variant(s) observed in base file, with: \n",
      "1998370 variant(s) excluded based on user input \n",
      "2006153 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "402175 variant(s) not found in previous data \n",
      "32089 ambiguous variant(s) excluded \n",
      "198884 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 40838 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  2  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-d-20405_0\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:37:22\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-d-20405_0.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-d-20405_0/ukb-d-20405_0.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-d-20405_0/ukb-d-20405_0 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 877111637 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-D-20405_0:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-d-20405_0 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-d-20405_0.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-D-20405_0:ES\tUKB-D-20405_0:SE\tUKB-D-20405_0:LP\tUKB-D-20405_0:AF\tUKB-D-20405_0:SS\tUKB-D-20405_0:EZ\tUKB-D-20405_0:SI\tUKB-D-20405_0:NC\tUKB-D-20405_0:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "13571659 variant(s) observed in base file, with: \n",
      "6766042 variant(s) excluded based on user input \n",
      "6805617 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "320277 variant(s) not found in previous data \n",
      "1 variant(s) with mismatch information \n",
      "44487 ambiguous variant(s) excluded \n",
      "268383 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62438 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  3  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-d-20405_1\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:38:08\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-d-20405_1.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-d-20405_1/ukb-d-20405_1.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-d-20405_1/ukb-d-20405_1 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2960688085 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-D-20405_1:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-d-20405_1 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-d-20405_1.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-D-20405_1:ES\tUKB-D-20405_1:SE\tUKB-D-20405_1:LP\tUKB-D-20405_1:AF\tUKB-D-20405_1:SS\tUKB-D-20405_1:EZ\tUKB-D-20405_1:SI\tUKB-D-20405_1:NC\tUKB-D-20405_1:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "12203570 variant(s) observed in base file, with: \n",
      "6126479 variant(s) excluded based on user input \n",
      "6077091 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "318737 variant(s) not found in previous data \n",
      "2 variant(s) with mismatch information \n",
      "44830 ambiguous variant(s) excluded \n",
      "269579 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62506 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  4  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-d-20405_2\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:38:50\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-d-20405_2.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-d-20405_2/ukb-d-20405_2.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-d-20405_2/ukb-d-20405_2 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2819275069 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-D-20405_2:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-d-20405_2 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-d-20405_2.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-D-20405_2:ES\tUKB-D-20405_2:SE\tUKB-D-20405_2:LP\tUKB-D-20405_2:AF\tUKB-D-20405_2:SS\tUKB-D-20405_2:EZ\tUKB-D-20405_2:SI\tUKB-D-20405_2:NC\tUKB-D-20405_2:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "12013476 variant(s) observed in base file, with: \n",
      "6064904 variant(s) excluded based on user input \n",
      "5948572 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "320242 variant(s) not found in previous data \n",
      "2 variant(s) with mismatch information \n",
      "44628 ambiguous variant(s) excluded \n",
      "268276 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62444 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  5  ------------------------------------\n",
      "*****GWAS ID IS***** met-d-Total_C\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:39:32\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/met-d-Total_C.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/met-d-Total_C/met-d-Total_C.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/met-d-Total_C/met-d-Total_C \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 429104453 \\\n",
      "    --snp ID \\\n",
      "    --stat MET-D-TOTAL_C:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing met-d-Total_C \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/met-d-Total_C.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tMET-D-TOTAL_C:ES\tMET-D-TOTAL_C:SE\tMET-D-TOTAL_C:LP\tMET-D-TOTAL_C:AF\tMET-D-TOTAL_C:SS\tMET-D-TOTAL_C:EZ\tMET-D-TOTAL_C:SI\tMET-D-TOTAL_C:NC\tMET-D-TOTAL_C:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "12216219 variant(s) observed in base file, with: \n",
      "623448 variant(s) excluded based on user input \n",
      "11592771 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5020 variant(s) not found in previous data \n",
      "89461 ambiguous variant(s) excluded \n",
      "538667 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92605 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  6  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-b-109\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:40:33\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-b-109.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ieu-b-109/ieu-b-109.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-b-109/ieu-b-109 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3730860376 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-B-109:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-b-109 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-b-109.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-B-109:ES\tIEU-B-109:SE\tIEU-B-109:LP\tIEU-B-109:AF\tIEU-B-109:SS\tIEU-B-109:EZ\tIEU-B-109:SI\tIEU-B-109:NC\tIEU-B-109:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "12321875 variant(s) observed in base file, with: \n",
      "370310 variant(s) excluded based on user input \n",
      "11951565 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "34109 variant(s) not found in previous data \n",
      "7 variant(s) with mismatch information \n",
      "85257 ambiguous variant(s) excluded \n",
      "513775 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 91152 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  7  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-b-110\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:41:36\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-b-110.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ieu-b-110/ieu-b-110.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-b-110/ieu-b-110 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 1809732829 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-B-110:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-b-110 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-b-110.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-B-110:ES\tIEU-B-110:SE\tIEU-B-110:LP\tIEU-B-110:AF\tIEU-B-110:SS\tIEU-B-110:EZ\tIEU-B-110:SI\tIEU-B-110:NC\tIEU-B-110:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "12321875 variant(s) observed in base file, with: \n",
      "370315 variant(s) excluded based on user input \n",
      "11951560 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "34106 variant(s) not found in previous data \n",
      "7 variant(s) with mismatch information \n",
      "85257 ambiguous variant(s) excluded \n",
      "513778 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 91242 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  8  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-b-111\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:42:39\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-b-111.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ieu-b-111/ieu-b-111.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-b-111/ieu-b-111 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 4267225795 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-B-111:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-b-111 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-b-111.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-B-111:ES\tIEU-B-111:SE\tIEU-B-111:LP\tIEU-B-111:AF\tIEU-B-111:SS\tIEU-B-111:EZ\tIEU-B-111:SI\tIEU-B-111:NC\tIEU-B-111:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "12321875 variant(s) observed in base file, with: \n",
      "370336 variant(s) excluded based on user input \n",
      "11951539 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "34108 variant(s) not found in previous data \n",
      "7 variant(s) with mismatch information \n",
      "85257 ambiguous variant(s) excluded \n",
      "513776 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 90997 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "2: 0\n",
      "-----------------------counter  9  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-b-25\n",
      "===== Done =====\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:43:42\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-b-25.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ieu-b-25/ieu-b-25.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-b-25/ieu-b-25 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 4211603569 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-B-25:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-b-25 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-b-25.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-B-25:ES\tIEU-B-25:SE\tIEU-B-25:LP\tIEU-B-25:AF\tIEU-B-25:SS\tIEU-B-25:EZ\tIEU-B-25:SI\tIEU-B-25:NC\tIEU-B-25:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "11913711 variant(s) observed in base file, with: \n",
      "70769 variant(s) excluded based on user input \n",
      "11842942 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "38371 variant(s) not found in previous data \n",
      "84232 ambiguous variant(s) excluded \n",
      "510545 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 91078 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "2: 0\n",
      "-----------------------counter  10  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-b-38\n",
      "===== Done =====\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:44:46\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-b-38.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ieu-b-38/ieu-b-38.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-b-38/ieu-b-38 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 461846946 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-B-38:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-b-38 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-b-38.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-B-38:ES\tIEU-B-38:SE\tIEU-B-38:LP\tIEU-B-38:AF\tIEU-B-38:SS\tIEU-B-38:EZ\tIEU-B-38:SI\tIEU-B-38:NC\tIEU-B-38:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "7088083 variant(s) observed in base file, with: \n",
      "37247 variant(s) excluded based on user input \n",
      "7050836 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "38616 variant(s) not found in previous data \n",
      "1 variant(s) with mismatch information \n",
      "84508 ambiguous variant(s) excluded \n",
      "510023 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 88422 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  11  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-b-39\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:45:26\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-b-39.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ieu-b-39/ieu-b-39.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-b-39/ieu-b-39 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3698562836 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-B-39:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-b-39 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-b-39.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-B-39:ES\tIEU-B-39:SE\tIEU-B-39:LP\tIEU-B-39:AF\tIEU-B-39:SS\tIEU-B-39:EZ\tIEU-B-39:SI\tIEU-B-39:NC\tIEU-B-39:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "7160619 variant(s) observed in base file, with: \n",
      "37676 variant(s) excluded based on user input \n",
      "7122943 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "37822 variant(s) not found in previous data \n",
      "1 variant(s) with mismatch information \n",
      "84646 ambiguous variant(s) excluded \n",
      "510679 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 88839 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  12  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-a-257\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:46:05\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-a-257.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-a-257/ukb-a-257 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 4287776523 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-A-257:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-a-257 \n",
      "================================================== \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-a-257.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-A-257:ES\tUKB-A-257:SE\tUKB-A-257:LP\tUKB-A-257:AF\tUKB-A-257:SS\tUKB-A-257:EZ\tUKB-A-257:SI\tUKB-A-257:NC\tUKB-A-257:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "10545186 variant(s) observed in base file, with: \n",
      "5257663 negative statistic observed. Maybe you have \n",
      "forgotten the --beta flag? \n",
      "5287523 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "320420 variant(s) not found in previous data \n",
      "200 variant(s) with mismatch information \n",
      "44362 ambiguous variant(s) excluded \n",
      "268166 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62390 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  13  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-12064\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:46:47\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-12064.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-12064/ukb-b-12064.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-12064/ukb-b-12064 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2271878681 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-12064:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-12064 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-12064.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-12064:ES\tUKB-B-12064:SE\tUKB-B-12064:LP\tUKB-B-12064:AF\tUKB-B-12064:SS\tUKB-B-12064:EZ\tUKB-B-12064:SI\tUKB-B-12064:NC\tUKB-B-12064:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "7338290 variant(s) observed in base file, with: \n",
      "3661731 variant(s) excluded based on user input \n",
      "3676559 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "318829 variant(s) not found in previous data \n",
      "45018 ambiguous variant(s) excluded \n",
      "269301 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62277 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  14  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-17627\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:47:13\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-17627.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-17627/ukb-b-17627.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-17627/ukb-b-17627 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2751861990 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-17627:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-17627 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-17627.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-17627:ES\tUKB-B-17627:SE\tUKB-B-17627:LP\tUKB-B-17627:AF\tUKB-B-17627:SS\tUKB-B-17627:EZ\tUKB-B-17627:SI\tUKB-B-17627:NC\tUKB-B-17627:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92586 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  15  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-18275\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:48:07\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-18275.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-18275/ukb-b-18275.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-18275/ukb-b-18275 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2099760720 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-18275:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-18275 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-18275.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-18275:ES\tUKB-B-18275:SE\tUKB-B-18275:LP\tUKB-B-18275:AF\tUKB-B-18275:SS\tUKB-B-18275:EZ\tUKB-B-18275:SI\tUKB-B-18275:NC\tUKB-B-18275:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9401286 variant(s) observed in base file, with: \n",
      "4713056 variant(s) excluded based on user input \n",
      "4688230 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "319448 variant(s) not found in previous data \n",
      "44682 ambiguous variant(s) excluded \n",
      "269018 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62543 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  16  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-19953\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:48:40\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-19953.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-19953/ukb-b-19953.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-19953/ukb-b-19953 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3973249766 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-19953:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-19953 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-19953.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-19953:ES\tUKB-B-19953:SE\tUKB-B-19953:LP\tUKB-B-19953:AF\tUKB-B-19953:SS\tUKB-B-19953:EZ\tUKB-B-19953:SI\tUKB-B-19953:NC\tUKB-B-19953:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92178 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  17  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-2209\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:49:33\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-2209.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-2209/ukb-b-2209.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-2209/ukb-b-2209 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2932852180 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-2209:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-2209 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-2209.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-2209:ES\tUKB-B-2209:SE\tUKB-B-2209:LP\tUKB-B-2209:AF\tUKB-B-2209:SS\tUKB-B-2209:EZ\tUKB-B-2209:SI\tUKB-B-2209:NC\tUKB-B-2209:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92530 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  18  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-3957\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:50:27\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-3957.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-3957/ukb-b-3957.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-3957/ukb-b-3957 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2992797737 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-3957:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-3957 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-3957.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-3957:ES\tUKB-B-3957:SE\tUKB-B-3957:LP\tUKB-B-3957:AF\tUKB-B-3957:SS\tUKB-B-3957:EZ\tUKB-B-3957:SI\tUKB-B-3957:NC\tUKB-B-3957:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92576 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  19  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-4424\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:51:21\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-4424.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-4424/ukb-b-4424.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-4424/ukb-b-4424 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2677015216 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-4424:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-4424 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-4424.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-4424:ES\tUKB-B-4424:SE\tUKB-B-4424:LP\tUKB-B-4424:AF\tUKB-B-4424:SS\tUKB-B-4424:EZ\tUKB-B-4424:SI\tUKB-B-4424:NC\tUKB-B-4424:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92386 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  20  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-6134\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:52:15\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-6134.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-6134/ukb-b-6134.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-6134/ukb-b-6134 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2227969776 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-6134:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-6134 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-6134.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-6134:ES\tUKB-B-6134:SE\tUKB-B-6134:LP\tUKB-B-6134:AF\tUKB-B-6134:SS\tUKB-B-6134:EZ\tUKB-B-6134:SI\tUKB-B-6134:NC\tUKB-B-6134:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9756889 variant(s) observed in base file, with: \n",
      "27793 variant(s) excluded based on user input \n",
      "9729096 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5024 variant(s) not found in previous data \n",
      "89456 ambiguous variant(s) excluded \n",
      "538668 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92598 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  21  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-6324\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:53:08\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-6324.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-6324/ukb-b-6324.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-6324/ukb-b-6324 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2963657595 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-6324:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-6324 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-6324.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-6324:ES\tUKB-B-6324:SE\tUKB-B-6324:LP\tUKB-B-6324:AF\tUKB-B-6324:SS\tUKB-B-6324:EZ\tUKB-B-6324:SI\tUKB-B-6324:NC\tUKB-B-6324:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92533 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  22  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-7663\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:54:01\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-7663.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-7663/ukb-b-7663.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-7663/ukb-b-7663 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 1626077631 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-7663:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-7663 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-7663.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-7663:ES\tUKB-B-7663:SE\tUKB-B-7663:LP\tUKB-B-7663:AF\tUKB-B-7663:SS\tUKB-B-7663:EZ\tUKB-B-7663:SI\tUKB-B-7663:NC\tUKB-B-7663:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "8164789 variant(s) observed in base file, with: \n",
      "4094164 variant(s) excluded based on user input \n",
      "4070625 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "319244 variant(s) not found in previous data \n",
      "44497 ambiguous variant(s) excluded \n",
      "269407 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62516 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  23  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-770\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:54:30\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-770.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-770/ukb-b-770.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-770/ukb-b-770 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 4281514655 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-770:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-770 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-770.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-770:ES\tUKB-B-770:SE\tUKB-B-770:LP\tUKB-B-770:AF\tUKB-B-770:SS\tUKB-B-770:EZ\tUKB-B-770:SI\tUKB-B-770:NC\tUKB-B-770:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "8625010 variant(s) observed in base file, with: \n",
      "14730 variant(s) excluded based on user input \n",
      "8610280 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5104 variant(s) not found in previous data \n",
      "89448 ambiguous variant(s) excluded \n",
      "538596 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92737 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "2: 0\n",
      "-----------------------counter  24  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-8476\n",
      "===== Done =====\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:55:20\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-8476.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-8476/ukb-b-8476.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-8476/ukb-b-8476 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 1225992808 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-8476:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-8476 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-8476.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-8476:ES\tUKB-B-8476:SE\tUKB-B-8476:LP\tUKB-B-8476:AF\tUKB-B-8476:SS\tUKB-B-8476:EZ\tUKB-B-8476:SI\tUKB-B-8476:NC\tUKB-B-8476:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "8862761 variant(s) observed in base file, with: \n",
      "4405830 variant(s) excluded based on user input \n",
      "4456931 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "317902 variant(s) not found in previous data \n",
      "44950 ambiguous variant(s) excluded \n",
      "270296 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62891 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  25  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-323\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:55:51\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-323.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-323/ukb-b-323.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-323/ukb-b-323 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3146239628 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-323:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-323 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-323.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-323:ES\tUKB-B-323:SE\tUKB-B-323:LP\tUKB-B-323:AF\tUKB-B-323:SS\tUKB-B-323:EZ\tUKB-B-323:SI\tUKB-B-323:NC\tUKB-B-323:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "6877801 variant(s) observed in base file, with: \n",
      "3455172 variant(s) excluded based on user input \n",
      "3422629 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "327203 variant(s) not found in previous data \n",
      "43653 ambiguous variant(s) excluded \n",
      "262292 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 60436 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "2: 0\n",
      "-----------------------counter  26  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-14699\n",
      "===== Done =====\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:56:16\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-14699.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-14699/ukb-b-14699.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-14699/ukb-b-14699 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 1677999424 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-14699:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-14699 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-14699.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-14699:ES\tUKB-B-14699:SE\tUKB-B-14699:LP\tUKB-B-14699:AF\tUKB-B-14699:SS\tUKB-B-14699:EZ\tUKB-B-14699:SI\tUKB-B-14699:NC\tUKB-B-14699:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "7793502 variant(s) observed in base file, with: \n",
      "3925276 variant(s) excluded based on user input \n",
      "3868226 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "321237 variant(s) not found in previous data \n",
      "44286 ambiguous variant(s) excluded \n",
      "267625 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62484 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  27  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-14180\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:56:43\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-14180.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-14180/ukb-b-14180.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-14180/ukb-b-14180 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2781494174 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-14180:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-14180 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-14180.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-14180:ES\tUKB-B-14180:SE\tUKB-B-14180:LP\tUKB-B-14180:AF\tUKB-B-14180:SS\tUKB-B-14180:EZ\tUKB-B-14180:SI\tUKB-B-14180:NC\tUKB-B-14180:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9515294 variant(s) observed in base file, with: \n",
      "4731485 variant(s) excluded based on user input \n",
      "4783809 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "317019 variant(s) not found in previous data \n",
      "45029 ambiguous variant(s) excluded \n",
      "271100 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62746 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  28  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-17243\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:57:16\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-17243.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-17243/ukb-b-17243.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-17243/ukb-b-17243 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2259262330 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-17243:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-17243 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-17243.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-17243:ES\tUKB-B-17243:SE\tUKB-B-17243:LP\tUKB-B-17243:AF\tUKB-B-17243:SS\tUKB-B-17243:EZ\tUKB-B-17243:SI\tUKB-B-17243:NC\tUKB-B-17243:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "5294159 variant(s) observed in base file, with: \n",
      "2655242 variant(s) excluded based on user input \n",
      "2638917 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "354265 variant(s) not found in previous data \n",
      "39313 ambiguous variant(s) excluded \n",
      "239570 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 52182 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  29  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-6358\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:57:35\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-6358.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-6358/ukb-b-6358.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-6358/ukb-b-6358 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 1824335327 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-6358:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-6358 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-6358.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-6358:ES\tUKB-B-6358:SE\tUKB-B-6358:LP\tUKB-B-6358:AF\tUKB-B-6358:SS\tUKB-B-6358:EZ\tUKB-B-6358:SI\tUKB-B-6358:NC\tUKB-B-6358:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "5221749 variant(s) observed in base file, with: \n",
      "2612651 variant(s) excluded based on user input \n",
      "2609098 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "355417 variant(s) not found in previous data \n",
      "39109 ambiguous variant(s) excluded \n",
      "238622 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 51395 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  30  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-17006\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:57:54\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-17006.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-17006/ukb-b-17006.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-17006/ukb-b-17006 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2328119855 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-17006:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-17006 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-17006.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-17006:ES\tUKB-B-17006:SE\tUKB-B-17006:LP\tUKB-B-17006:AF\tUKB-B-17006:SS\tUKB-B-17006:EZ\tUKB-B-17006:SI\tUKB-B-17006:NC\tUKB-B-17006:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "2541600 variant(s) observed in base file, with: \n",
      "1278228 variant(s) excluded based on user input \n",
      "1263372 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "479374 variant(s) not found in previous data \n",
      "20301 ambiguous variant(s) excluded \n",
      "133473 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 28554 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  31  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-5779\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:58:06\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-5779.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-5779/ukb-b-5779.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-5779/ukb-b-5779 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 935388042 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-5779:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-5779 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-5779.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-5779:ES\tUKB-B-5779:SE\tUKB-B-5779:LP\tUKB-B-5779:AF\tUKB-B-5779:SS\tUKB-B-5779:EZ\tUKB-B-5779:SI\tUKB-B-5779:NC\tUKB-B-5779:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92454 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  32  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-15541\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:58:59\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-15541.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-15541/ukb-b-15541.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-15541/ukb-b-15541 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2613811686 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-15541:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-15541 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-15541.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-15541:ES\tUKB-B-15541:SE\tUKB-B-15541:LP\tUKB-B-15541:AF\tUKB-B-15541:SS\tUKB-B-15541:EZ\tUKB-B-15541:SI\tUKB-B-15541:NC\tUKB-B-15541:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "4801285 variant(s) observed in base file, with: \n",
      "2403231 variant(s) excluded based on user input \n",
      "2398054 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "370617 variant(s) not found in previous data \n",
      "36962 ambiguous variant(s) excluded \n",
      "225569 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 47933 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  33  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-19732\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:59:17\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-19732.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-19732/ukb-b-19732.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-19732/ukb-b-19732 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2413711990 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-19732:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-19732 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-19732.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-19732:ES\tUKB-B-19732:SE\tUKB-B-19732:LP\tUKB-B-19732:AF\tUKB-B-19732:SS\tUKB-B-19732:EZ\tUKB-B-19732:SI\tUKB-B-19732:NC\tUKB-B-19732:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "7112787 variant(s) observed in base file, with: \n",
      "3559403 variant(s) excluded based on user input \n",
      "3553384 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "322235 variant(s) not found in previous data \n",
      "44165 ambiguous variant(s) excluded \n",
      "266748 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 61031 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  34  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-20289\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:59:42\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-20289.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-20289/ukb-b-20289.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-20289/ukb-b-20289 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3106944125 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-20289:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-20289 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-20289.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-20289:ES\tUKB-B-20289:SE\tUKB-B-20289:LP\tUKB-B-20289:AF\tUKB-B-20289:SS\tUKB-B-20289:EZ\tUKB-B-20289:SI\tUKB-B-20289:NC\tUKB-B-20289:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "4329979 variant(s) observed in base file, with: \n",
      "2168808 variant(s) excluded based on user input \n",
      "2161171 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "388671 variant(s) not found in previous data \n",
      "34240 ambiguous variant(s) excluded \n",
      "210237 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 43700 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  35  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-14057\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 19:59:58\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-14057.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-14057/ukb-b-14057.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-14057/ukb-b-14057 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2666811341 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-14057:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-14057 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-14057.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-14057:ES\tUKB-B-14057:SE\tUKB-B-14057:LP\tUKB-B-14057:AF\tUKB-B-14057:SS\tUKB-B-14057:EZ\tUKB-B-14057:SI\tUKB-B-14057:NC\tUKB-B-14057:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9154199 variant(s) observed in base file, with: \n",
      "4590066 variant(s) excluded based on user input \n",
      "4564133 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "320315 variant(s) not found in previous data \n",
      "44917 ambiguous variant(s) excluded \n",
      "267916 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62212 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  36  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-12963\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:00:30\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-12963.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-12963/ukb-b-12963.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-12963/ukb-b-12963 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3841396074 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-12963:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-12963 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-12963.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-12963:ES\tUKB-B-12963:SE\tUKB-B-12963:LP\tUKB-B-12963:AF\tUKB-B-12963:SS\tUKB-B-12963:EZ\tUKB-B-12963:SI\tUKB-B-12963:NC\tUKB-B-12963:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92681 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  37  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-12417\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:01:22\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-12417.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-12417/ukb-b-12417.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-12417/ukb-b-12417 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 4286879231 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-12417:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-12417 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-12417.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-12417:ES\tUKB-B-12417:SE\tUKB-B-12417:LP\tUKB-B-12417:AF\tUKB-B-12417:SS\tUKB-B-12417:EZ\tUKB-B-12417:SI\tUKB-B-12417:NC\tUKB-B-12417:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9851866 variant(s) observed in base file, with: \n",
      "29450 variant(s) excluded based on user input \n",
      "9822416 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5018 variant(s) not found in previous data \n",
      "89457 ambiguous variant(s) excluded \n",
      "538673 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92554 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  38  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-11495\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:02:15\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-11495.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-11495/ukb-b-11495.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-11495/ukb-b-11495 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2783394386 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-11495:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-11495 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-11495.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-11495:ES\tUKB-B-11495:SE\tUKB-B-11495:LP\tUKB-B-11495:AF\tUKB-B-11495:SS\tUKB-B-11495:EZ\tUKB-B-11495:SI\tUKB-B-11495:NC\tUKB-B-11495:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9606172 variant(s) observed in base file, with: \n",
      "4877388 variant(s) excluded based on user input \n",
      "4728784 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "322896 variant(s) not found in previous data \n",
      "43987 ambiguous variant(s) excluded \n",
      "266265 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 61885 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  39  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-12493\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:02:47\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-12493.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-12493/ukb-b-12493.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-12493/ukb-b-12493 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 45995175 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-12493:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-12493 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-12493.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-12493:ES\tUKB-B-12493:SE\tUKB-B-12493:LP\tUKB-B-12493:AF\tUKB-B-12493:SS\tUKB-B-12493:EZ\tUKB-B-12493:SI\tUKB-B-12493:NC\tUKB-B-12493:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "8361394 variant(s) observed in base file, with: \n",
      "4197604 variant(s) excluded based on user input \n",
      "4163790 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "319808 variant(s) not found in previous data \n",
      "44681 ambiguous variant(s) excluded \n",
      "268659 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62384 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  40  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-4282\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:03:17\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-4282.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-4282/ukb-b-4282.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-4282/ukb-b-4282 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 723770088 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-4282:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-4282 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-4282.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-4282:ES\tUKB-B-4282:SE\tUKB-B-4282:LP\tUKB-B-4282:AF\tUKB-B-4282:SS\tUKB-B-4282:EZ\tUKB-B-4282:SI\tUKB-B-4282:NC\tUKB-B-4282:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9331187 variant(s) observed in base file, with: \n",
      "20908 variant(s) excluded based on user input \n",
      "9310279 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "5040 variant(s) not found in previous data \n",
      "89455 ambiguous variant(s) excluded \n",
      "538653 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92650 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "2: 0\n",
      "-----------------------counter  41  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-4171\n",
      "stat {'or'}\n",
      "stat {'or'}\n",
      "===== Done =====\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:04:08\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-4171.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-4171/ukb-b-4171.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-4171/ukb-b-4171 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3491966556 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-4171:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-4171 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-4171.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-4171:ES\tUKB-B-4171:SE\tUKB-B-4171:LP\tUKB-B-4171:AF\tUKB-B-4171:SS\tUKB-B-4171:EZ\tUKB-B-4171:SI\tUKB-B-4171:NC\tUKB-B-4171:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9188063 variant(s) observed in base file, with: \n",
      "4621864 variant(s) excluded based on user input \n",
      "4566199 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "320656 variant(s) not found in previous data \n",
      "44464 ambiguous variant(s) excluded \n",
      "268028 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62428 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  42  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-4000\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:04:39\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-4000.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-4000/ukb-b-4000.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-4000/ukb-b-4000 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 974217197 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-4000:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-4000 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-4000.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-4000:ES\tUKB-B-4000:SE\tUKB-B-4000:LP\tUKB-B-4000:AF\tUKB-B-4000:SS\tUKB-B-4000:EZ\tUKB-B-4000:SI\tUKB-B-4000:NC\tUKB-B-4000:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9256599 variant(s) observed in base file, with: \n",
      "4656377 variant(s) excluded based on user input \n",
      "4600222 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "319322 variant(s) not found in previous data \n",
      "44411 ambiguous variant(s) excluded \n",
      "269415 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62427 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  43  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-d-20544_1\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:05:13\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-d-20544_1.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-d-20544_1/ukb-d-20544_1.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-d-20544_1/ukb-d-20544_1 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2849695841 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-D-20544_1:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-d-20544_1 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-d-20544_1.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-D-20544_1:ES\tUKB-D-20544_1:SE\tUKB-D-20544_1:LP\tUKB-D-20544_1:AF\tUKB-D-20544_1:SS\tUKB-D-20544_1:EZ\tUKB-D-20544_1:SI\tUKB-D-20544_1:NC\tUKB-D-20544_1:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "9938650 variant(s) observed in base file, with: \n",
      "5014255 variant(s) excluded based on user input \n",
      "4924395 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "319561 variant(s) not found in previous data \n",
      "1 variant(s) with mismatch information \n",
      "44654 ambiguous variant(s) excluded \n",
      "268932 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62283 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  44  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-d-20544_15\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:05:48\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-d-20544_15.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-d-20544_15/ukb-d-20544_15.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-d-20544_15/ukb-d-20544_15 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 4102699046 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-D-20544_15:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-d-20544_15 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-d-20544_15.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-D-20544_15:ES\tUKB-D-20544_15:SE\tUKB-D-20544_15:LP\tUKB-D-20544_15:AF\tUKB-D-20544_15:SS\tUKB-D-20544_15:EZ\tUKB-D-20544_15:SI\tUKB-D-20544_15:NC\tUKB-D-20544_15:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "13571546 variant(s) observed in base file, with: \n",
      "6815432 variant(s) excluded based on user input \n",
      "6756114 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "320600 variant(s) not found in previous data \n",
      "44577 ambiguous variant(s) excluded \n",
      "267971 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62316 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  45  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-20556\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:06:35\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-20556.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-20556/ukb-b-20556.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-20556/ukb-b-20556 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3670259693 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-20556:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-20556 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-20556.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-20556:ES\tUKB-B-20556:SE\tUKB-B-20556:LP\tUKB-B-20556:AF\tUKB-B-20556:SS\tUKB-B-20556:EZ\tUKB-B-20556:SI\tUKB-B-20556:NC\tUKB-B-20556:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "3084288 variant(s) observed in base file, with: \n",
      "1540822 variant(s) excluded based on user input \n",
      "1543466 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "448206 variant(s) not found in previous data \n",
      "25028 ambiguous variant(s) excluded \n",
      "159914 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 32939 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  46  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-d-20441\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:06:47\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-d-20441.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-d-20441/ukb-d-20441.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-d-20441/ukb-d-20441 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 919907872 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-D-20441:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-d-20441 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-d-20441.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-D-20441:ES\tUKB-D-20441:SE\tUKB-D-20441:LP\tUKB-D-20441:AF\tUKB-D-20441:SS\tUKB-D-20441:EZ\tUKB-D-20441:SI\tUKB-D-20441:NC\tUKB-D-20441:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "13571567 variant(s) observed in base file, with: \n",
      "6783247 variant(s) excluded based on user input \n",
      "6788320 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "318536 variant(s) not found in previous data \n",
      "1 variant(s) with mismatch information \n",
      "44945 ambiguous variant(s) excluded \n",
      "269666 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 62395 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  47  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-18162\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:07:33\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-18162.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-18162/ukb-b-18162.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-18162/ukb-b-18162 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 2225356039 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-18162:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-18162 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-18162.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-18162:ES\tUKB-B-18162:SE\tUKB-B-18162:LP\tUKB-B-18162:AF\tUKB-B-18162:SS\tUKB-B-18162:EZ\tUKB-B-18162:SI\tUKB-B-18162:NC\tUKB-B-18162:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "8020737 variant(s) observed in base file, with: \n",
      "4006420 variant(s) excluded based on user input \n",
      "4014317 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "317479 variant(s) not found in previous data \n",
      "44932 ambiguous variant(s) excluded \n",
      "270737 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 63034 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  48  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-14624\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:08:01\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-14624.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-14624/ukb-b-14624.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-14624/ukb-b-14624 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3399114632 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-14624:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-14624 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-14624.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-14624:ES\tUKB-B-14624:SE\tUKB-B-14624:LP\tUKB-B-14624:AF\tUKB-B-14624:SS\tUKB-B-14624:EZ\tUKB-B-14624:SI\tUKB-B-14624:NC\tUKB-B-14624:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "6444208 variant(s) observed in base file, with: \n",
      "3203114 variant(s) excluded based on user input \n",
      "3241094 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "327603 variant(s) not found in previous data \n",
      "43315 ambiguous variant(s) excluded \n",
      "262230 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 59444 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  49  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-b-9667\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:08:24\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-b-9667.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-b-9667/ukb-b-9667.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-b-9667/ukb-b-9667 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 1611065233 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-B-9667:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-b-9667 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-b-9667.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-B-9667:ES\tUKB-B-9667:SE\tUKB-B-9667:LP\tUKB-B-9667:AF\tUKB-B-9667:SS\tUKB-B-9667:EZ\tUKB-B-9667:SI\tUKB-B-9667:NC\tUKB-B-9667:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "8338738 variant(s) observed in base file, with: \n",
      "4122753 variant(s) excluded based on user input \n",
      "4215985 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "314279 variant(s) not found in previous data \n",
      "45391 ambiguous variant(s) excluded \n",
      "273478 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 63588 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  50  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-a-1183\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:08:54\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-a-1183.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-a-1183/ieu-a-1183 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 701420322 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-A-1183:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-a-1183 \n",
      "================================================== \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-a-1183.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-A-1183:ES\tIEU-A-1183:SE\tIEU-A-1183:LP\tIEU-A-1183:AF\tIEU-A-1183:SS\tIEU-A-1183:EZ\tIEU-A-1183:SI\tIEU-A-1183:NC\tIEU-A-1183:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "7468461 variant(s) observed in base file, with: \n",
      "13868 NA stat/p-value observed \n",
      "3646992 negative statistic observed. Maybe you have \n",
      "forgotten the --beta flag? \n",
      "3807601 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "334995 variant(s) not found in previous data \n",
      "8 variant(s) with mismatch information \n",
      "42388 ambiguous variant(s) excluded \n",
      "255757 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 55551 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  51  ------------------------------------\n",
      "*****GWAS ID IS***** ukb-d-30500_irnt\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:09:23\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ukb-d-30500_irnt.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ukb-d-30500_irnt/ukb-d-30500_irnt.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ukb-d-30500_irnt/ukb-d-30500_irnt \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3487873943 \\\n",
      "    --snp ID \\\n",
      "    --stat UKB-D-30500_IRNT:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ukb-d-30500_irnt \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ukb-d-30500_irnt.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tUKB-D-30500_IRNT:ES\tUKB-D-30500_IRNT:SE\tUKB-D-30500_IRNT:LP\tUKB-D-30500_IRNT:AF\tUKB-D-30500_IRNT:SS\tUKB-D-30500_IRNT:EZ\tUKB-D-30500_IRNT:SI\tUKB-D-30500_IRNT:NC\tUKB-D-30500_IRNT:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "13571116 variant(s) observed in base file, with: \n",
      "25132 variant(s) excluded based on user input \n",
      "13545984 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "6664 variant(s) not found in previous data \n",
      "3 variant(s) with mismatch information \n",
      "89322 ambiguous variant(s) excluded \n",
      "537159 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 92015 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  52  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-b-7\n",
      "stat {'or'}\n",
      "stat {'or'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:10:34\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-b-7.tsv \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ieu-b-7/ieu-b-7.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --or  \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-b-7/ieu-b-7 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3960629183 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-B-7:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-b-7 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-b-7.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-B-7:ES\tIEU-B-7:SE\tIEU-B-7:LP\tIEU-B-7:AF\tIEU-B-7:SS\tIEU-B-7:EZ\tIEU-B-7:SI\tIEU-B-7:NC\tIEU-B-7:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "17891936 variant(s) observed in base file, with: \n",
      "9876285 variant(s) excluded based on user input \n",
      "8015651 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "337300 variant(s) not found in previous data \n",
      "100 variant(s) with mismatch information \n",
      "42091 ambiguous variant(s) excluded \n",
      "253657 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 61628 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s) with p-value between 0.1 and 1e-5 \n",
      "(\u001b[1;31mmay not be significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n",
      "-----------------------counter  53  ------------------------------------\n",
      "*****GWAS ID IS***** ieu-b-5067\n",
      "stat {'beta'}\n",
      "stat {'beta'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-08-21 20:11:30\n",
      "../PRSice_linux/PRSice_linux \\\n",
      "    --a1 ALT \\\n",
      "    --a2 Z \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base ../tabular_format_gwas_data/ieu-b-5067.tsv \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --cov covar_affymetrix.txt \\\n",
      "    --extract ./PRSice_output_ROSMAP_2/ieu-b-5067/ieu-b-5067.valid \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --num-auto 22 \\\n",
      "    --out ./PRSice_output_ROSMAP_2/ieu-b-5067/ieu-b-5067 \\\n",
      "    --pvalue PVAL_generated_from_LP \\\n",
      "    --score std \\\n",
      "    --seed 3534305514 \\\n",
      "    --snp ID \\\n",
      "    --stat IEU-B-5067:ES \\\n",
      "    --target rosmap_affymetrix \\\n",
      "    --thread 2 \\\n",
      "    --upper 0.5\n",
      "\n",
      "\u001b[1;33mWarning: X not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Z not found in base file \u001b[0m\n",
      "\u001b[1;33mWarning: Y not found in base file \u001b[0m\n",
      "\n",
      "Initializing Genotype file: rosmap_affymetrix (bed) \n",
      "\n",
      "Start processing ieu-b-5067 \n",
      "================================================== \n",
      "\n",
      "SNP extraction/exclusion list contains 4 columns, will \n",
      "assume first column contains the SNP ID \n",
      "\n",
      "Base file: ../tabular_format_gwas_data/ieu-b-5067.tsv \n",
      "Header of file is: \n",
      "\n",
      "CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO:AF\tIEU-B-5067:ES\tIEU-B-5067:SE\tIEU-B-5067:LP\tIEU-B-5067:AF\tIEU-B-5067:SS\tIEU-B-5067:EZ\tIEU-B-5067:SI\tIEU-B-5067:NC\tIEU-B-5067:ID\tPVAL_generated_from_LP \n",
      "\n",
      "Reading 100.00%\n",
      "12321875 variant(s) observed in base file, with: \n",
      "370291 variant(s) excluded based on user input \n",
      "11951584 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "1644 people (953 male(s), 670 female(s)) observed \n",
      "1644 founder(s) included \n",
      "\n",
      "34105 variant(s) not found in previous data \n",
      "7 variant(s) with mismatch information \n",
      "85257 ambiguous variant(s) excluded \n",
      "513779 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 91507 \n",
      "\n",
      "Processing the 1 th phenotype \n",
      "\n",
      "Phenotype is a continuous phenotype \n",
      "1644 sample(s) with valid phenotype \n",
      "\n",
      "Processing the covariate file: covar_affymetrix.txt \n",
      "============================== \n",
      "\n",
      "Include Covariates: \n",
      "Name\tMissing\tNumber of levels \n",
      "C1\t0\t- \n",
      "C2\t0\t- \n",
      "C3\t0\t- \n",
      "C4\t0\t- \n",
      "C5\t0\t- \n",
      "C6\t0\t- \n",
      "C7\t0\t- \n",
      "C8\t0\t- \n",
      "C9\t0\t- \n",
      "C10\t0\t- \n",
      "\n",
      "After reading the covariate file, 1644 sample(s) included \n",
      "in the analysis \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "There are 1 region(s)/phenotype(s) with p-value > 0.1 (\u001b[1;31mnot \n",
      "significant\u001b[0m). \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Plotting the quantile plot\n",
      "Plotting Bar Plot\n",
      "Plotting the high resolution plot\n",
      "1: 0\n",
      "===== Done =====\n",
      "2: 0\n"
     ]
    }
   ],
   "source": [
    "# here for some files a weird error occurs: \"reporter not initialized\"\n",
    "# soln : create corresponding directory\n",
    "# bbj-a-78\n",
    "# ukb-a-257\n",
    "# bbj-a-46\n",
    "# these are not included due to not being in the EUR ancestry\n",
    "counter = 0\n",
    "for GWAS_ID in base_files:\n",
    "#     if GWAS_ID == 'ukb-b-4282':\n",
    "    counter += 1\n",
    "    \n",
    "    print('-----------------------counter ',counter,' ------------------------------------')\n",
    "    print('*****GWAS ID IS*****',GWAS_ID)\n",
    "#         run_prsice(GWAS_ID=GWAS_ID)\n",
    "    run_prsice_new(GWAS_ID=GWAS_ID)\n",
    "\n",
    "#         problem -> they do not have corresponding entries in prsice_output folder->create those folders manually"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "base_files\n",
    "GWAS_ID = 'ukb-b-14057'\n",
    "\n",
    "#====\n",
    "\n",
    "def get_prs_values(GWAS_ID):\n",
    "    prsice_output = f'./PRSice_output_ROSMAP_2/{GWAS_ID}/'\n",
    "    prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy() \n",
    "    return prs_array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1644"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_prs_values(GWAS_ID=GWAS_ID).shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "base files {'ukb-b-13806': 'or', 'ukb-d-20405_0': 'or', 'ukb-d-20405_1': 'or', 'ukb-d-20405_2': 'or', 'met-d-Total_C': 'beta', 'ieu-b-109': 'beta', 'ieu-b-110': 'beta', 'ieu-b-111': 'beta', 'ieu-b-25': 'beta', 'ieu-b-38': 'beta', 'ieu-b-39': 'beta', 'ukb-a-257': 'or', 'ukb-b-12064': 'or', 'ukb-b-17627': 'beta', 'ukb-b-18275': 'or', 'ukb-b-19953': 'beta', 'ukb-b-2209': 'beta', 'ukb-b-3957': 'beta', 'ukb-b-4424': 'beta', 'ukb-b-6134': 'beta', 'ukb-b-6324': 'beta', 'ukb-b-7663': 'or', 'ukb-b-770': 'beta', 'ukb-b-8476': 'or', 'ukb-b-323': 'or', 'ukb-b-14699': 'or', 'ukb-b-14180': 'or', 'ukb-b-17243': 'or', 'ukb-b-6358': 'or', 'ukb-b-17006': 'or', 'ukb-b-5779': 'beta', 'ukb-b-15541': 'or', 'ukb-b-19732': 'or', 'ukb-b-20289': 'or', 'ukb-b-14057': 'or', 'ukb-b-12963': 'beta', 'ukb-b-12417': 'beta', 'ukb-b-11495': 'or', 'ukb-b-12493': 'or', 'ukb-b-4282': 'beta', 'ukb-b-4171': 'or', 'ukb-b-4000': 'or', 'ukb-d-20544_1': 'or', 'ukb-d-20544_15': 'or', 'ukb-b-20556': 'or', 'ukb-d-20441': 'or', 'ukb-b-18162': 'or', 'ukb-b-14624': 'or', 'ukb-b-9667': 'or', 'ieu-a-1183': 'or', 'ukb-d-30500_irnt': 'beta', 'ieu-b-7': 'or', 'ieu-b-5067': 'beta'}\n",
      "new base files ['ukb-b-13806', 'ukb-d-20405_0', 'ukb-d-20405_1', 'ukb-d-20405_2', 'met-d-Total_C', 'ieu-b-109', 'ieu-b-110', 'ieu-b-111', 'ieu-b-25', 'ieu-b-38', 'ieu-b-39', 'ukb-a-257', 'ukb-b-12064', 'ukb-b-17627', 'ukb-b-18275', 'ukb-b-19953', 'ukb-b-2209', 'ukb-b-3957', 'ukb-b-4424', 'ukb-b-6134', 'ukb-b-6324', 'ukb-b-7663', 'ukb-b-770', 'ukb-b-8476', 'ukb-b-323', 'ukb-b-14699', 'ukb-b-14180', 'ukb-b-17243', 'ukb-b-6358', 'ukb-b-17006', 'ukb-b-5779', 'ukb-b-15541', 'ukb-b-19732', 'ukb-b-20289', 'ukb-b-14057', 'ukb-b-12963', 'ukb-b-12417', 'ukb-b-11495', 'ukb-b-12493', 'ukb-b-4282', 'ukb-b-4171', 'ukb-b-4000', 'ukb-d-20544_1', 'ukb-d-20544_15', 'ukb-b-20556', 'ukb-d-20441', 'ukb-b-18162', 'ukb-b-14624', 'ukb-b-9667', 'ieu-a-1183', 'ukb-d-30500_irnt', 'ieu-b-7', 'ieu-b-5067']\n",
      "ukb-b-13806\n",
      "ukb-d-20405_0\n",
      "ukb-d-20405_1\n",
      "ukb-d-20405_2\n",
      "met-d-Total_C\n",
      "ieu-b-109\n",
      "ieu-b-110\n",
      "ieu-b-111\n",
      "ieu-b-25\n",
      "ieu-b-38\n",
      "ieu-b-39\n",
      "ukb-a-257\n",
      "ukb-b-12064\n",
      "ukb-b-17627\n",
      "ukb-b-18275\n",
      "ukb-b-19953\n",
      "ukb-b-2209\n",
      "ukb-b-3957\n",
      "ukb-b-4424\n",
      "ukb-b-6134\n",
      "ukb-b-6324\n",
      "ukb-b-7663\n",
      "ukb-b-770\n",
      "ukb-b-8476\n",
      "ukb-b-323\n",
      "ukb-b-14699\n",
      "ukb-b-14180\n",
      "ukb-b-17243\n",
      "ukb-b-6358\n",
      "ukb-b-17006\n",
      "ukb-b-5779\n",
      "ukb-b-15541\n",
      "ukb-b-19732\n",
      "ukb-b-20289\n",
      "ukb-b-14057\n",
      "ukb-b-12963\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ukb-b-12417\n",
      "ukb-b-11495\n",
      "ukb-b-12493\n",
      "ukb-b-4282\n",
      "ukb-b-4171\n",
      "ukb-b-4000\n",
      "ukb-d-20544_1\n",
      "ukb-d-20544_15\n",
      "ukb-b-20556\n",
      "ukb-d-20441\n",
      "ukb-b-18162\n",
      "ukb-b-14624\n",
      "ukb-b-9667\n",
      "ieu-a-1183\n",
      "ukb-d-30500_irnt\n",
      "ieu-b-7\n",
      "ieu-b-5067\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n",
      "/tmp/ipykernel_25612/173594717.py:8: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os.path\n",
    "import json\n",
    "name_mapping={}\n",
    "# GWAS_ID\n",
    "# ukb-b-13806\n",
    "# ukb-b-12064\n",
    "# ukb-b-323\n",
    "# ukb-b-14699\n",
    "# ukb-b-5779\n",
    "# base_files=base_files-\n",
    "print('base files',base_files)\n",
    "newBaseFiles=[]\n",
    "for GWAS_ID in base_files:\n",
    "    prsice_output = f'./PRSice_output_ROSMAP_2/{GWAS_ID}/{GWAS_ID}.best'\n",
    "    if os.path.exists(prsice_output):  \n",
    "        newBaseFiles.append(GWAS_ID)\n",
    "print('new base files',newBaseFiles)\n",
    "\n",
    "PRS_feature_matrix = np.zeros([len(newBaseFiles), 1644])\n",
    "# PRS_feature_matrix = np.zeros([len(newBaseFiles), 846])\n",
    "\n",
    "for i, GWAS_ID in enumerate(newBaseFiles):\n",
    "    print(GWAS_ID)\n",
    "    prsice_output = f'./PRSice_output_ROSMAP_2/{GWAS_ID}/{GWAS_ID}.best'\n",
    "    if os.path.exists(prsice_output):  \n",
    "        PRS_feature_matrix[i, :] = get_prs_values(GWAS_ID=GWAS_ID)\n",
    "        name_mapping[GWAS_ID] =df[df['GWAS_ID'] == GWAS_ID].to_numpy()[0, 1]\n",
    "    else :\n",
    "        print(\"no path exists\")\n",
    "        \n",
    "# with open(inputFolder+\"traits_map.json\", \"w\") as outfile:\n",
    "with open(\"traits_map_with_AD.json\", \"w\") as outfile:\n",
    "    \n",
    "    json.dump(name_mapping, outfile)\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[-0.9168994  -0.21442193  0.95399238 ...  0.57118296  1.63892612\n",
      "   0.69484085]\n",
      " [ 0.38457489  0.22569779  0.53657    ... -0.04018068  0.05085179\n",
      "  -1.36606973]\n",
      " [ 1.19558689 -0.87952628  0.74057583 ... -0.31038049 -0.84914888\n",
      "  -0.27851194]\n",
      " ...\n",
      " [ 0.99177941  2.33325315  0.55505037 ...  0.52680986 -0.52338171\n",
      "   0.82470505]\n",
      " [-2.36913759  0.02526208  1.86836562 ...  0.21903392  0.05202987\n",
      "   0.89417101]\n",
      " [-2.96444071  1.04745396 -0.27104461 ...  0.21018242  0.82667241\n",
      "  -0.23211876]]\n"
     ]
    }
   ],
   "source": [
    "print(PRS_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "PRS_feature_matrix = PRS_feature_matrix.T\n",
    "np.save('PRS_feature_matrix_with_ad', PRS_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1644, 53)\n"
     ]
    }
   ],
   "source": [
    "print(PRS_feature_matrix.shape)\n",
    "import random"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Only with AD PRS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_17813/1739813349.py:21: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  prs_array = pd.read_csv(prsice_output+f'{GWAS_ID}.best', ' ')['PRS'].to_numpy()\n"
     ]
    }
   ],
   "source": [
    "# PRS_feature_matrix_only_ad = np.zeros([1, 1816])\n",
    "PRS_feature_matrix_only_ad = get_prs_values(GWAS_ID='ieu-b-5067')\n",
    "PRS_feature_matrix_only_ad = PRS_feature_matrix_only_ad.T\n",
    "np.save('PRS_feature_matrix_only_ad', PRS_feature_matrix_only_ad)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "550 1094 1644\n"
     ]
    }
   ],
   "source": [
    "TRAIN_SAMPLE_SIZE = 550\n",
    "train_samples = list(range(1644))\n",
    "random.shuffle(train_samples)\n",
    "train_samples, test_samples = train_samples[:TRAIN_SAMPLE_SIZE], train_samples[TRAIN_SAMPLE_SIZE:]\n",
    "# print(test_samples)\n",
    "print(len(train_samples), len(test_samples), len(train_samples) + len(test_samples))\n",
    "# PRS_feature_matrix[train_samples].mean(0), PRS_feature_matrix[train_samples].std(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([-3.20063684e-03, -1.16745630e-02, -1.62563695e-03, -1.45995396e-02,\n",
       "         1.81226160e-02, -1.41222658e-02,  2.59397513e-02, -1.08676569e-03,\n",
       "        -1.59275442e-02,  1.06056163e-02,  1.00998294e-03, -1.12513684e-02,\n",
       "         1.19402802e-03,  5.46455829e-03, -1.19946661e-02, -1.64919659e-02,\n",
       "         1.28068050e-02, -9.79679559e-03,  1.47792065e-02,  2.43447114e-04,\n",
       "        -1.99630757e-03, -1.09297553e-02, -1.91106013e-02, -7.10920847e-04,\n",
       "        -1.03662225e-02, -1.91184444e-02, -9.33204446e-03, -2.12532227e-02,\n",
       "        -2.91836350e-02, -6.25583773e-03, -1.99435276e-02,  4.13992541e-04,\n",
       "        -1.07788069e-02,  7.99917727e-03, -1.64618723e-02, -1.66578261e-02,\n",
       "        -1.64490084e-02, -5.95392853e-03, -1.83166027e-02, -2.12051609e-02,\n",
       "        -3.29552667e-03, -1.06555431e-02, -1.63715148e-02, -8.72369661e-03,\n",
       "         6.65254670e-03,  6.00976296e-03, -1.49823886e-02,  3.32215118e-07,\n",
       "         7.59193063e-03, -2.53701504e-04]),\n",
       " array([0.95372266, 0.94400142, 0.97165727, 0.90807297, 0.9794307 ,\n",
       "        1.02487039, 0.9965353 , 0.99606001, 0.99052919, 1.00104849,\n",
       "        1.00524695, 0.9288827 , 0.99299107, 0.99095256, 0.93342815,\n",
       "        0.98910239, 1.01753085, 1.00687874, 0.9022691 , 0.99760172,\n",
       "        1.00030945, 0.91534946, 0.9885595 , 0.97777315, 0.99099556,\n",
       "        0.99082316, 0.97022882, 0.97408027, 0.97823395, 0.99108851,\n",
       "        0.8904033 , 0.99865293, 0.90653442, 0.99919717, 0.93600575,\n",
       "        0.9032774 , 0.91601003, 0.91479259, 0.92699031, 0.94898975,\n",
       "        0.94062901, 0.94073395, 0.90553653, 0.91585274, 0.99220891,\n",
       "        0.99843672, 0.9578753 , 0.98522894, 0.99398665, 1.01698892]))"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PRS_feature_matrix[test_samples].mean(0), PRS_feature_matrix[test_samples].std(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([ 1.80121253e-11, -7.23843658e-11, -4.03398744e-11, -1.24741255e-11,\n",
       "         1.19856782e-11,  8.02313086e-12, -3.06668484e-11,  6.21563858e-11,\n",
       "        -2.72191628e-11,  7.99063703e-12,  7.19383295e-11, -1.15251910e-12,\n",
       "        -3.74664176e-11,  1.85110097e-11, -5.24669764e-11,  1.11084739e-11,\n",
       "         5.59416361e-11,  3.21943862e-11, -5.24559427e-11, -2.76734093e-12,\n",
       "        -1.95044098e-11,  9.72282264e-12,  5.42070353e-12,  6.10582832e-11,\n",
       "         3.02257739e-11,  1.54763251e-11,  3.52919065e-11, -6.37114248e-12,\n",
       "        -8.16188233e-12,  4.53524247e-11, -1.82294982e-11,  3.02863363e-12,\n",
       "         6.98566804e-12,  3.05832070e-11, -1.58479201e-13,  4.16432195e-11,\n",
       "         2.62714803e-11,  1.76151057e-11, -4.53576677e-12]),\n",
       " array([0.99972463, 0.99972463, 0.99972463, 0.99972463, 0.99972463,\n",
       "        0.99972463, 0.99972463, 0.99972463, 0.99972463, 0.99972463,\n",
       "        0.99972463, 0.99972463, 0.99972463, 0.99972463, 0.99972463,\n",
       "        0.99972463, 0.99972463, 0.99972463, 0.99972463, 0.99972463,\n",
       "        0.99972463, 0.99972463, 0.99972463, 0.99972463, 0.99972463,\n",
       "        0.99972463, 0.99972463, 0.99972463, 0.99972463, 0.99972463,\n",
       "        0.99972463, 0.99972463, 0.99972463, 0.99972463, 0.99972463,\n",
       "        0.99972463, 0.99972463, 0.99972463, 0.99972463]))"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PRS_feature_matrix[:].mean(0), PRS_feature_matrix[:].std(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'ROS20275399': 0, 'ROS10442701': 1, 'ROS20152393': 2, 'ROS20626558': 3, 'ROS15176592': 4, 'ROS21174691': 5, 'ROS15146050': 6, 'ROS10606903': 7, 'ROS20969088': 8, 'ROS11319389': 9, 'ROS21185180': 10, 'ROS10433070': 11, 'ROS10439580': 12, 'ROS10490993': 13, 'ROS20266118': 14, 'ROS11444465': 15, 'ROS21001933': 16, 'ROS20984415': 17, 'ROS20706215': 18, 'ROS21112011': 19, 'ROS20297791': 20, 'ROS21411297': 21, 'ROS20550027': 22, 'ROS11342432': 23, 'ROS10481686': 24, 'ROS15179365': 25, 'ROS20998065': 26, 'ROS20327084': 27, 'ROS11395417': 28, 'ROS10478041': 29, 'ROS10277308': 30, 'ROS20993308': 31, 'ROS21117192': 32, 'ROS20996621': 33, 'ROS20730959': 34, 'ROS21405627': 35, 'ROS20966477': 36, 'ROS20594669': 37, 'ROS20611993': 38, 'ROS10533083': 39, 'ROS10460587': 40, 'ROS20500815': 41, 'ROS21142003': 42, 'ROS20946545': 43, 'ROS10460299': 44, 'ROS20956605': 45, 'ROS20521752': 46, 'ROS20339740': 47, 'ROS21181988': 48, 'ROS11259716': 49, 'ROS20128523': 50, 'ROS20177982': 51, 'ROS10572120': 52, 'ROS21174303': 53, 'ROS21001807': 54, 'ROS10669174': 55, 'ROS20102505': 56, 'ROS20487369': 57, 'ROS10479920': 58, 'ROS11430815': 59, 'ROS11310844': 60, 'ROS10208143': 61, 'ROS10536568': 62, 'ROS11697592': 63, 'ROS15121461': 64, 'ROS20603141': 65, 'ROS21172121': 66, 'ROS20174533': 67, 'ROS15113169': 68, 'ROS20956867': 69, 'ROS21117516': 70, 'ROS20610726': 71, 'ROS10850171': 72, 'ROS21412626': 73, 'ROS20431359': 74, 'ROS20665307': 75, 'ROS20376029': 76, 'ROS15420223': 77, 'ROS21401399': 78, 'ROS10235948': 79, 'ROS15114174': 80, 'ROS10684501': 81, 'ROS10417638': 82, 'ROS20288196': 83, 'ROS20240514': 84, 'ROS20171508': 85, 'ROS20969114': 86, 'ROS10249336': 87, 'ROS20970441': 88, 'ROS21000504': 89, 'ROS11327005': 90, 'ROS21193068': 91, 'ROS10101291': 92, 'ROS20929774': 93, 'ROS20660226': 94, 'ROS10485914': 95, 'ROS11349119': 96, 'ROS15797073': 97, 'ROS20233939': 98, 'ROS11460357': 99, 'ROS20156469': 100, 'ROS15483818': 101, 'ROS20311676': 102, 'ROS15144878': 103, 'ROS11467746': 104, 'ROS10502798': 105, 'ROS20906493': 106, 'ROS20164059': 107, 'ROS10202345': 108, 'ROS20236838': 109, 'ROS20602974': 110, 'ROS20141228': 111, 'ROS20139850': 112, 'ROS20195344': 113, 'ROS20147026': 114, 'ROS20506325': 115, 'ROS21402016': 116, 'ROS21000766': 117, 'ROS20249897': 118, 'ROS21101658': 119, 'ROS10929637': 120, 'ROS10290427': 121, 'ROS20299973': 122, 'ROS11345331': 123, 'ROS11157783': 124, 'ROS20492186': 125, 'ROS21411323': 126, 'ROS21402304': 127, 'ROS15110134': 128, 'ROS15115927': 129, 'ROS21126085': 130, 'ROS10976490': 131, 'ROS20115276': 132, 'ROS20904509': 133, 'ROS21191200': 134, 'ROS21246218': 135, 'ROS20584923': 136, 'ROS20774339': 137, 'ROS20559562': 138, 'ROS15139925': 139, 'ROS20557128': 140, 'ROS20561642': 141, 'ROS20532115': 142, 'ROS20530195': 143, 'ROS21232244': 144, 'ROS20232186': 145, 'ROS11302830': 146, 'ROS11072071': 147, 'ROS20106383': 148, 'ROS20234818': 149, 'ROS61215831': 150, 'ROS20925446': 151, 'ROS20219931': 152, 'ROS21140119': 153, 'ROS20909680': 154, 'ROS20197364': 155, 'ROS20105242': 156, 'ROS20130577': 157, 'ROS11344452': 158, 'ROS20990085': 159, 'ROS21276174': 160, 'ROS15147641': 161, 'ROS20494368': 162, 'ROS10308444': 163, 'ROS20225051': 164, 'ROS20151388': 165, 'ROS15702926': 166, 'ROS20474724': 167, 'ROS10510090': 168, 'ROS21189544': 169, 'ROS21144735': 170, 'ROS21131352': 171, 'ROS21002524': 172, 'ROS21155350': 173, 'ROS10986136': 174, 'ROS20270768': 175, 'ROS10425642': 176, 'ROS11200645': 177, 'ROS21403157': 178, 'ROS20390027': 179, 'ROS10490579': 180, 'ROS11069724': 181, 'ROS20630946': 182, 'ROS11335695': 183, 'ROS21274866': 184, 'ROS21176585': 185, 'ROS21002812': 186, 'ROS10275964': 187, 'ROS21000630': 188, 'ROS20683921': 189, 'ROS20133764': 190, 'ROS10255456': 191, 'ROS20416210': 192, 'ROS20726273': 193, 'ROS10660639': 194, 'ROS20224046': 195, 'ROS11448667': 196, 'ROS10499302': 197, 'ROS21236608': 198, 'ROS11165535': 199, 'ROS10291856': 200, 'ROS20865035': 201, 'ROS21120287': 202, 'MAP50197261': 203, 'MAP50104684': 204, 'MAP31037310': 205, 'MAP51826377': 206, 'MAP61344957': 207, 'MAP50305165': 208, 'MAP50105725': 209, 'MAP90447310': 210, 'MAP50100482': 211, 'MAP85666580': 212, 'MAP90214403': 213, 'MAP50107583': 214, 'MAP50108174': 215, 'MAP89001223': 216, 'MAP50300408': 217, 'MAP22813029': 218, 'MAP50401976': 219, 'MAP70883578': 220, 'MAP62483327': 221, 'MAP94722642': 222, 'MAP84642424': 223, 'MAP70726480': 224, 'MAP23870000': 225, 'MAP93462021': 226, 'MAP43074402': 227, 'MAP50302978': 228, 'MAP50102240': 229, 'MAP29933130': 230, 'MAP77143621': 231, 'MAP15218541': 232, 'MAP50401264': 233, 'MAP50100518': 234, 'MAP92113025': 235, 'MAP50103967': 236, 'MAP25300551': 237, 'MAP82684406': 238, 'MAP61142759': 239, 'MAP95116046': 240, 'MAP50409406': 241, 'MAP22868024': 242, 'MAP50101659': 243, 'MAP56696733': 244, 'MAP50104134': 245, 'MAP50104972': 246, 'MAP36492755': 247, 'MAP50302004': 248, 'MAP12365619': 249, 'MAP07883803': 250, 'MAP51815338': 251, 'MAP95330358': 252, 'MAP50410319': 253, 'MAP09391376': 254, 'MAP83034844': 255, 'MAP72188804': 256, 'MAP50100932': 257, 'MAP09650662': 258, 'MAP50102952': 259, 'MAP50302716': 260, 'MAP34999847': 261, 'MAP50102376': 262, 'MAP08109170': 263, 'MAP07253015': 264, 'MAP50103381': 265, 'MAP05498462': 266, 'MAP31509843': 267, 'MAP50404749': 268, 'MAP50409370': 269, 'MAP50408815': 270, 'MAP52509923': 271, 'MAP50301251': 272, 'MAP66924745': 273, 'MAP50302554': 274, 'MAP67216828': 275, 'MAP74078795': 276, 'MAP50407936': 277, 'MAP54458919': 278, 'MAP56102646': 279, 'MAP50102088': 280, 'MAP34962204': 281, 'MAP35941263': 282, 'MAP50406057': 283, 'MAP45566083': 284, 'MAP91018909': 285, 'MAP50410283': 286, 'MAP87779516': 287, 'MAP61029627': 288, 'MAP53389845': 289, 'MAP50400709': 290, 'MAP50301413': 291, 'MAP50301387': 292, 'MAP50108886': 293, 'MAP03713990': 294, 'MAP82317494': 295, 'MAP04879843': 296, 'MAP65001607': 297, 'MAP50109189': 298, 'MAP37178462': 299, 'MAP17929065': 300, 'MAP76008327': 301, 'MAP37452211': 302, 'MAP25580771': 303, 'MAP46369736': 304, 'MAP64216968': 305, 'MAP92629514': 306, 'MAP50402017': 307, 'MAP60848460': 308, 'MAP50300084': 309, 'MAP92996364': 310, 'MAP01243685': 311, 'MAP26498416': 312, 'MAP70625336': 313, 'MAP72650337': 314, 'MAP80187972': 315, 'MAP81852640': 316, 'MAP53772202': 317, 'MAP09841821': 318, 'MAP00482428': 319, 'MAP64493027': 320, 'MAP93787649': 321, 'MAP71727427': 322, 'MAP50107907': 323, 'MAP41773404': 324, 'MAP68015667': 325, 'MAP98953007': 326, 'MAP50402431': 327, 'MAP50106316': 328, 'ROS21180423': 329, 'ROS20280666': 330, 'ROS20239863': 331, 'ROS20965598': 332, 'ROS20260284': 333, 'ROS20178159': 334, 'ROS20905352': 335, 'ROS10516762': 336, 'ROS10345561': 337, 'ROS20372539': 338, 'ROS21001069': 339, 'ROS21155774': 340, 'ROS20170917': 341, 'ROS21114031': 342, 'ROS20223455': 343, 'ROS20264936': 344, 'ROS20254902': 345, 'ROS10494519': 346, 'ROS10383017': 347, 'ROS20951100': 348, 'ROS20312979': 349, 'ROS10495100': 350, 'ROS10359111': 351, 'ROS21183160': 352, 'ROS11216503': 353, 'ROS20917694': 354, 'ROS10442413': 355, 'ROS20261037': 356, 'ROS15148520': 357, 'ROS20178447': 358, 'ROS15143863': 359, 'ROS20344143': 360, 'ROS11632851': 361, 'ROS20557542': 362, 'ROS10462057': 363, 'ROS20643329': 364, 'ROS11313031': 365, 'ROS20954261': 366, 'ROS20654818': 367, 'ROS11353057': 368, 'ROS11464261': 369, 'ROS20267411': 370, 'ROS20244554': 371, 'ROS15461128': 372, 'ROS15138884': 373, 'ROS20252720': 374, 'ROS11624423': 375, 'ROS10288185': 376, 'ROS42874904': 377, 'ROS41817635': 378, 'MAP66406040': 379, 'ROS11118510': 380, 'MAP50104710': 381, 'MAP50101497': 382, 'MAP60747316': 383, 'MAP91921829': 384, 'MAP71626373': 385, 'MAP50301675': 386, 'MAP50407800': 387, 'MAP50103129': 388, 'MAP17219510': 389, 'MAP32689956': 390, 'MAP50104558': 391, 'MAP02817047': 392, 'MAP05779736': 393, 'MAP43485807': 394, 'MAP47067421': 395, 'MAP05689621': 396, 'MAP50406471': 397, 'MAP60871487': 398, 'MAP69982533': 399, 'MAP82624422': 400, 'MAP86471999': 401, 'MAP50400835': 402, 'MAP35286551': 403, 'MAP70816595': 404, 'MAP85065193': 405, 'MAP39316600': 406, 'MAP67185070': 407, 'MAP71648351': 408, 'MAP06073025': 409, 'MAP30755009': 410, 'MAP50101073': 411, 'MAP36010337': 412, 'MAP02899847': 413, 'MAP86071821': 414, 'MAP14498577': 415, 'MAP45450112': 416, 'MAP23690880': 417, 'MAP50301549': 418, 'MAP27560777': 419, 'MAP64291829': 420, 'MAP97006937': 421, 'MAP53682197': 422, 'MAP50102664': 423, 'MAP06107196': 424, 'MAP50302266': 425, 'MAP50402567': 426, 'MAP50408941': 427, 'MAP82527318': 428, 'MAP50403310': 429, 'MAP77970662': 430, 'MAP50106578': 431, 'MAP50302428': 432, 'MAP50105563': 433, 'MAP95491648': 434, 'MAP93815598': 435, 'MAP80333458': 436, 'MAP45115248': 437, 'MAP90942860': 438, 'MAP04947970': 439, 'MAP50105987': 440, 'MAP62404688': 441, 'MAP08132197': 442, 'MAP24039289': 443, 'MAP24310553': 444, 'MAP36830117': 445, 'MAP50102114': 446, 'MAP50104846': 447, 'MAP50105275': 448, 'MAP50106730': 449, 'MAP50106992': 450, 'MAP50107169': 451, 'MAP50107745': 452, 'MAP50300110': 453, 'MAP50404163': 454, 'MAP54122640': 455, 'MAP57978756': 456, 'MAP65652206': 457, 'MAP90267190': 458, 'MAP92613520': 459, 'ROS11030909': 460, 'ROS20271359': 461, 'ROS10203224': 462, 'ROS10514454': 463, 'ROS11440263': 464, 'ROS20187042': 465, 'ROS20800682': 466, 'ROS10537609': 467, 'ROS20149910': 468, 'ROS20232474': 469, 'ROS11616707': 470, 'ROS11631558': 471, 'ROS15413198': 472, 'ROS20906079': 473, 'ROS21176459': 474, 'ROS11466579': 475, 'ROS10253148': 476, 'ROS21407511': 477, 'ROS20787136': 478, 'ROS10205244': 479, 'ROS11471260': 480, 'ROS20767628': 481, 'ROS11455530': 482, 'ROS20112377': 483, 'ROS21232118': 484, 'ROS11409232': 485, 'ROS11413170': 486, 'ROS11310718': 487, 'ROS21000180': 488, 'ROS20201189': 489, 'ROS20914669': 490, 'ROS11275158': 491, 'ROS20221273': 492, 'ROS11229148': 493, 'ROS11173549': 494, 'ROS10435090': 495, 'ROS10524640': 496, 'ROS10901987': 497, 'ROS20963578': 498, 'ROS11681508': 499, 'ROS15844425': 500, 'ROS20242958': 501, 'ROS15773499': 502, 'ROS20953832': 503, 'ROS10480383': 504, 'ROS20165190': 505, 'ROS20108141': 506, 'ROS21411459': 507, 'ROS10885370': 508, 'ROS20185860': 509, 'ROS20798913': 510, 'ROS20124321': 511, 'ROS20148879': 512, 'ROS20179164': 513, 'ROS21157370': 514, 'ROS10424475': 515, 'ROS20945666': 516, 'MAP22776575': 517, 'MAP64358179': 518, 'MAP09032080': 519, 'MAP50102826': 520, 'MAP31756046': 521, 'MAP87264456': 522, 'MAP86903794': 523, 'MAP77891596': 524, 'MAP66754397': 525, 'MAP00668310': 526, 'ROS21156815': 527, 'ROS15712086': 528, 'ROS20267709': 529, 'MAP50104422': 530, 'MAP50100806': 531, 'MAP46000440': 532, 'MAP36379623': 533, 'MAP61490006': 534, 'MAP50301099': 535, 'ROS86094238': 536, 'ROS10292311': 537, 'ROS10222853': 538, 'ROS11336574': 539, 'ROS21159840': 540, 'ROS20194465': 541, 'ROS11421184': 542, 'ROS10419072': 543, 'ROS10300914': 544, 'ROS20201927': 545, 'ROS20143248': 546, 'ROS21125818': 547, 'ROS10101327': 548, 'ROS10218339': 549, 'ROS15495024': 550, 'ROS20104101': 551, 'ROS10271474': 552, 'ROS11606935': 553, 'ROS20780035': 554, 'ROS21158709': 555, 'MAP51624179': 556, 'MAP33411712': 557, 'MAP39226595': 558, 'MAP36267530': 559, 'MAP96892753': 560, 'MAP67531158': 561, 'MAP69432088': 562, 'MAP68725248': 563, 'MAP95453354': 564, 'ROS20939708': 565, 'ROS21002236': 566, 'MAP50410157': 567, 'MAP57833358': 568, 'ROS20153272': 569, 'MAP28314511': 570, 'ROS15831042': 571, 'MAP23601829': 572, 'ROS21156103': 573, 'MAP50409820': 574, 'ROS11390174': 575, 'ROS21192189': 576, 'ROS21406344': 577, 'ROS20173942': 578, 'ROS20667589': 579, 'MAP75862131': 580, 'ROS20646778': 581, 'MAP72830557': 582, 'ROS20895027': 583, 'MAP69544171': 584, 'ROS20738353': 585, 'ROS20877791': 586, 'ROS21162197': 587, 'ROS20794261': 588, 'MAP91347203': 589, 'MAP78511269': 590, 'ROS21160753': 591, 'ROS20739782': 592, 'MAP71514280': 593, 'MAP78123271': 594, 'MAP66787314': 595, 'ROS20834164': 596, 'ROS20574151': 597, 'ROS20631825': 598, 'ROS20779672': 599, 'MAP67831051': 600, 'ROS21175732': 601, 'ROS20723662': 602, 'ROS20555072': 603, 'MAP67628490': 604, 'MAP72777797': 605, 'ROS20559986': 606, 'ROS20608934': 607, 'ROS91698297': 608, 'MAP91707643': 609, 'ROS21144023': 610, 'ROS94716946': 611, 'ROS88922125': 612, 'ROS86495735': 613, 'MAP91349809': 614, 'MAP97775875': 615, 'ROS64777164': 616, 'ROS59796318': 617, 'MAP97130008': 618, 'ROS67123238': 619, 'ROS57737709': 620, 'MAP95281012': 621, 'ROS21128131': 622, 'MAP90155007': 623, 'MAP92308774': 624, 'MAP90434801': 625, 'ROS62775766': 626, 'MAP95868799': 627, 'ROS87268108': 628, 'ROS77098163': 629, 'MAP91897758': 630, 'ROS75142238': 631, 'MAP16322424': 632, 'ROS48213489': 633, 'ROS21135130': 634, 'ROS68671371': 635, 'ROS21119086': 636, 'ROS85186331': 637, 'MAP92393245': 638, 'ROS21135554': 639, 'ROS47071819': 640, 'MAP97220113': 641, 'MAP98322204': 642, 'MAP94115009': 643, 'ROS21138327': 644, 'ROS54474801': 645, 'MAP90544686': 646, 'ROS59407454': 647, 'MAP97882751': 648, 'ROS97929179': 649, 'MAP94430339': 650, 'MAP68778359': 651, 'MAP75495116': 652, 'MAP77239958': 653, 'ROS20603727': 654, 'ROS20662822': 655, 'ROS20535564': 656, 'ROS20848390': 657, 'ROS20821943': 658, 'MAP72659872': 659, 'ROS20712597': 660, 'MAP76867532': 661, 'ROS21180847': 662, 'MAP69577198': 663, 'MAP77330002': 664, 'ROS20618256': 665, 'ROS20616100': 666, 'ROS20649389': 667, 'ROS20402408': 668, 'MAP76733461': 669, 'MAP72560222': 670, 'MAP70669392': 671, 'MAP73549281': 672, 'MAP72076711': 673, 'ROS20431061': 674, 'ROS20739944': 675, 'MAP73369061': 676, 'MAP66800446': 677, 'MAP98506202': 678, 'ROS21135680': 679, 'ROS21123474': 680, 'ROS59371363': 681, 'MAP97749823': 682, 'ROS43462940': 683, 'ROS64858250': 684, 'ROS75005810': 685, 'ROS94593487': 686, 'MAP91280115': 687, 'MAP94144536': 688, 'MAP99367207': 689, 'MAP96140000': 690, 'ROS98627990': 691, 'ROS56603444': 692, 'MAP97264179': 693, 'MAP93103023': 694, 'ROS73026243': 695, 'ROS48453395': 696, 'MAP90977195': 697, 'MAP94362202': 698, 'ROS83211453': 699, 'ROS56265406': 700, 'ROS86072386': 701, 'MAP16513683': 702, 'ROS70421820': 703, 'MAP91189485': 704, 'ROS21137312': 705, 'MAP69331934': 706, 'ROS21172257': 707, 'MAP65539462': 708, 'MAP72324068': 709, 'ROS20886846': 710, 'MAP71063018': 711, 'ROS20506451': 712, 'MAP72347501': 713, 'MAP78195885': 714, 'ROS20549538': 715, 'MAP74753465': 716, 'MAP68914513': 717, 'MAP94205114': 718, 'ROS21133372': 719, 'ROS21125782': 720, 'MAP99809748': 721, 'ROS60269051': 722, 'ROS21274604': 723, 'MAP81795552': 724, 'ROS20940035': 725, 'ROS20567738': 726, 'MAP74494179': 727, 'ROS20399274': 728, 'ROS20202806': 729, 'ROS20204114': 730, 'MAP50103093': 731, 'MAP50104260': 732, 'MAP12799845': 733, 'ROS20260860': 734, 'ROS20371236': 735, 'MAP63740337': 736, 'MAP40401703': 737, 'MAP14696573': 738, 'MAP06881437': 739, 'MAP07117738': 740, 'ROS20331760': 741, 'MAP08637647': 742, 'MAP07142937': 743, 'ROS20355470': 744, 'ROS20224208': 745, 'MAP15286377': 746, 'ROS20177306': 747, 'MAP80606718': 748, 'MAP90021936': 749, 'ROS20966927': 750, 'ROS20971294': 751, 'MAP86037865': 752, 'ROS20907246': 753, 'ROS20921794': 754, 'ROS21001221': 755, 'MAP87330987': 756, 'MAP84653463': 757, 'ROS20995742': 758, 'ROS21002498': 759, 'MAP79938680': 760, 'MAP83001827': 761, 'MAP82643501': 762, 'MAP79590228': 763, 'ROS20972759': 764, 'ROS20911508': 765, 'ROS20936223': 766, 'ROS20954009': 767, 'ROS20897173': 768, 'MAP81874628': 769, 'MAP89546375': 770, 'MAP82325110': 771, 'MAP80592317': 772, 'ROS21113576': 773, 'ROS20959766': 774, 'ROS20959216': 775, 'MAP78905055': 776, 'MAP89968269': 777, 'MAP79680333': 778, 'MAP85171938': 779, 'ROS21103542': 780, 'ROS20912225': 781, 'MAP81086436': 782, 'ROS20986985': 783, 'MAP88106709': 784, 'MAP84949680': 785, 'MAP83984043': 786, 'MAP81526371': 787, 'ROS20968235': 788, 'MAP85925902': 789, 'MAP87208386': 790, 'ROS20952377': 791, 'ROS21104269': 792, 'MAP89512181': 793, 'ROS21001195': 794, 'ROS21001519': 795, 'MAP83721144': 796, 'ROS20949570': 797, 'ROS20973638': 798, 'ROS21001771': 799, 'MAP79501267': 800, 'ROS21002650': 801, 'ROS20974193': 802, 'MAP88499289': 803, 'MAP86880662': 804, 'ROS20940611': 805, 'ROS20980187': 806, 'ROS20950807': 807, 'MAP79692387': 808, 'ROS20938117': 809, 'ROS20898476': 810, 'ROS21002100': 811, 'MAP87195990': 812, 'ROS20281833': 813, 'ROS20300213': 814, 'MAP64150173': 815, 'MAP50410021': 816, 'MAP09986931': 817, 'MAP15302308': 818, 'MAP13380780': 819, 'MAP14074375': 820, 'MAP15387421': 821, 'ROS20207437': 822, 'MAP64145770': 823, 'ROS20254290': 824, 'MAP50103417': 825, 'MAP64336939': 826, 'ROS20278324': 827, 'ROS20282974': 828, 'ROS20368565': 829, 'ROS20187204': 830, 'ROS20276430': 831, 'ROS20380831': 832, 'MAP09508427': 833, 'ROS20261901': 834, 'MAP12317427': 835, 'MAP50101361': 836, 'MAP40071481': 837, 'ROS20254326': 838, 'MAP39989287': 839, 'ROS21293107': 840, 'MAP15373321': 841, 'ROS20254740': 842, 'ROS20191566': 843, 'ROS20173104': 844, 'ROS20248206': 845, 'ROS20208992': 846, 'ROS21298738': 847, 'ROS20294342': 848, 'ROS21400410': 849, 'ROS20219643': 850, 'MAP14397423': 851, 'ROS20201891': 852, 'ROS20258528': 853, 'ROS20222990': 854, 'MAP31813134': 855, 'MAP39800111': 856, 'ROS20254164': 857, 'MAP80345114': 858, 'MAP85509528': 859, 'ROS20935218': 860, 'MAP84858308': 861, 'ROS20946121': 862, 'MAP89614402': 863, 'ROS21111718': 864, 'MAP87942151': 865, 'MAP15376482': 866, 'ROS20260734': 867, 'MAP13346850': 868, 'MAP64287431': 869, 'MAP31824173': 870, 'ROS20297403': 871, 'ROS20921532': 872, 'MAP39393581': 873, 'MAP38618915': 874, 'MAP38113465': 875, 'ROS20682204': 876, 'ROS20964295': 877, 'ROS20732429': 878, 'MAP31989791': 879, 'MAP37865636': 880, 'ROS20596265': 881, 'ROS21123050': 882, 'MAP39125441': 883, 'ROS20113418': 884, 'MAP37280555': 885, 'ROS20769648': 886, 'MAP33781047': 887, 'ROS20988167': 888, 'ROS20973890': 889, 'MAP38056377': 890, 'MAP39282539': 891, 'ROS20934889': 892, 'MAP36128228': 893, 'MAP39680713': 894, 'MAP32863530': 895, 'MAP34781079': 896, 'ROS21002074': 897, 'ROS20610140': 898, 'ROS20268976': 899, 'ROS11389397': 900, 'MAP56751351': 901, 'MAP63144733': 902, 'ROS21406920': 903, 'ROS18932470': 904, 'MAP50502020': 905, 'MAP58677906': 906, 'ROS11605894': 907, 'MAP57597479': 908, 'ROS21408526': 909, 'ROS11455792': 910, 'MAP59306274': 911, 'ROS11699350': 912, 'MAP55177253': 913, 'ROS15111013': 914, 'ROS11436875': 915, 'MAP51442191': 916, 'MAP58306828': 917, 'MAP59618707': 918, 'ROS21408652': 919, 'ROS11661126': 920, 'MAP61389630': 921, 'ROS11466605': 922, 'ROS14817724': 923, 'ROS15470309': 924, 'MAP55932428': 925, 'ROS16718942': 926, 'ROS20120679': 927, 'MAP55296057': 928, 'MAP61579172': 929, 'MAP51129205': 930, 'ROS11669232': 931, 'MAP58501637': 932, 'ROS11659334': 933, 'MAP52178858': 934, 'MAP51650969': 935, 'ROS21108911': 936, 'ROS20947550': 937, 'ROS21404450': 938, 'MAP36773029': 939, 'MAP36087310': 940, 'MAP37527863': 941, 'ROS20551582': 942, 'MAP37251992': 943, 'MAP35847634': 944, 'ROS20741150': 945, 'ROS20936061': 946, 'MAP39541825': 947, 'MAP37439930': 948, 'ROS21406768': 949, 'MAP35079112': 950, 'ROS21404748': 951, 'MAP33118460': 952, 'ROS21123636': 953, 'ROS20997924': 954, 'ROS21000054': 955, 'MAP32244817': 956, 'MAP36751041': 957, 'ROS20923002': 958, 'ROS20925996': 959, 'ROS20954847': 960, 'ROS20927880': 961, 'MAP36951779': 962, 'ROS20986435': 963, 'MAP39484737': 964, 'MAP37190440': 965, 'ROS20892128': 966, 'ROS20779708': 967, 'MAP32705437': 968, 'MAP36997205': 969, 'MAP36953249': 970, 'MAP31914288': 971, 'ROS21104395': 972, 'ROS20634274': 973, 'ROS21145740': 974, 'ROS20153858': 975, 'ROS20110221': 976, 'MAP57978468': 977, 'ROS15108504': 978, 'ROS15778868': 979, 'MAP53533437': 980, 'ROS15166270': 981, 'MAP51400993': 982, 'ROS15780948': 983, 'ROS11656273': 984, 'MAP60922329': 985, 'MAP55820335': 986, 'MAP58199939': 987, 'ROS11630679': 988, 'ROS15436055': 989, 'MAP60961592': 990, 'ROS11444591': 991, 'ROS11649850': 992, 'ROS11392518': 993, 'MAP59150662': 994, 'MAP53200779': 995, 'ROS11434729': 996, 'ROS15857934': 997, 'ROS20131870': 998, 'MAP59756266': 999, 'ROS15178486': 1000, 'ROS21407935': 1001, 'ROS15763627': 1002, 'ROS20162877': 1003, 'ROS20139562': 1004, 'ROS11475462': 1005, 'ROS11651930': 1006, 'ROS20130865': 1007, 'MAP33679954': 1008, 'ROS20976799': 1009, 'ROS20613887': 1010, 'ROS20819999': 1011, 'ROS19621010': 1012, 'MAP17615774': 1013, 'ROS15196848': 1014, 'MAP41357606': 1015, 'MAP41285665': 1016, 'MAP48013463': 1017, 'ROS10100312': 1018, 'MAP46178577': 1019, 'MAP49766153': 1020, 'ROS22540345': 1021, 'MAP46547648': 1022, 'ROS21213589': 1023, 'MAP48399494': 1024, 'ROS10101165': 1025, 'MAP44655449': 1026, 'MAP17974060': 1027, 'MAP17255334': 1028, 'ROS15103261': 1029, 'ROS11241702': 1030, 'ROS10101589': 1031, 'ROS15175325': 1032, 'ROS15161351': 1033, 'MAP30261598': 1034, 'ROS10470809': 1035, 'ROS11258675': 1036, 'MAP30819298': 1037, 'MAP27054288': 1038, 'ROS11446647': 1039, 'ROS10405008': 1040, 'MAP31533749': 1041, 'ROS02495739': 1042, 'ROS10488101': 1043, 'MAP31587865': 1044, 'MAP25974431': 1045, 'MAP27481601': 1046, 'ROS11092291': 1047, 'ROS10233504': 1048, 'ROS10438999': 1049, 'ROS10506864': 1050, 'MAP27865599': 1051, 'ROS21214206': 1052, 'ROS10149161': 1053, 'ROS21171080': 1054, 'MAP47214624': 1055, 'MAP40611041': 1056, 'MAP50100644': 1057, 'MAP47371712': 1058, 'ROS21172545': 1059, 'MAP45984063': 1060, 'MAP46291609': 1061, 'MAP49070444': 1062, 'ROS10101615': 1063, 'ROS21213453': 1064, 'MAP48451825': 1065, 'MAP42988567': 1066, 'ROS31908032': 1067, 'MAP42433067': 1068, 'ROS10100574': 1069, 'MAP42023091': 1070, 'ROS23538323': 1071, 'MAP45986371': 1072, 'MAP18112988': 1073, 'MAP19429100': 1074, 'ROS10222303': 1075, 'ROS15182748': 1076, 'MAP17233356': 1077, 'ROS11408227': 1078, 'ROS11233086': 1079, 'ROS15196424': 1080, 'ROS11619606': 1081, 'MAP22865387': 1082, 'MAP18455382': 1083, 'ROS11371707': 1084, 'MAP17260313': 1085, 'MAP22644848': 1086, 'ROS11655268': 1087, 'ROS10221262': 1088, 'MAP29306403': 1089, 'ROS11612505': 1090, 'MAP22408684': 1091, 'ROS15706704': 1092, 'ROS21213877': 1093, 'MAP27908391': 1094, 'ROS11159965': 1095, 'MAP25423683': 1096, 'MAP47315778': 1097, 'MAP43191312': 1098, 'MAP42949394': 1099, 'MAP44751386': 1100, 'ROS10100862': 1101, 'ROS10100448': 1102, 'ROS21191624': 1103, 'ROS21402728': 1104, 'ROS10202507': 1105, 'MAP44842532': 1106, 'ROS29481089': 1107, 'ROS10100736': 1108, 'ROS10101741': 1109, 'ROS95939725': 1110, 'ROS11384054': 1111, 'ROS15707133': 1112, 'MAP23892088': 1113, 'MAP29540773': 1114, 'MAP44985334': 1115, 'MAP49676048': 1116, 'MAP41296604': 1117, 'ROS04169712': 1118, 'ROS21153754': 1119, 'ROS07299965': 1120, 'ROS27427687': 1121, 'MAP42387756': 1122, 'ROS21412590': 1123, 'ROS10101039': 1124, 'ROS10100600': 1125, 'ROS88941880': 1126, 'MAP46516939': 1127, 'MAP40622080': 1128, 'MAP44472870': 1129, 'MAP42297641': 1130, 'ROS10202633': 1131, 'ROS01237015': 1132, 'ROS29803009': 1133, 'MAP47236602': 1134, 'MAP46189516': 1135, 'ROS10100286': 1136, 'ROS53575211': 1137, 'ROS10007374': 1138, 'MAP47652086': 1139, 'ROS10200901': 1140, 'ROS06608444': 1141, 'ROS70760860': 1142, 'ROS11207908': 1143, 'ROS11469504': 1144, 'MAP19325445': 1145, 'MAP27313862': 1146, 'ROS15141429': 1147, 'MAP17413576': 1148, 'ROS10246987': 1149, 'MAP22486606': 1150, 'MAP17827744': 1151, 'MAP23791808': 1152, 'ROS10266369': 1153, 'ROS10284019': 1154, 'ROS10319069': 1155, 'ROS10100998': 1156, 'ROS10205532': 1157, 'MAP30311425': 1158, 'ROS10686233': 1159, 'ROS10102206': 1160, 'ROS21274578': 1161, 'ROS20114009': 1162, 'MAP05102083': 1163, 'MAP68611261': 1164, 'ROS20540417': 1165, 'MAP99110004': 1166, 'ROS10395061': 1167, 'ROS11166252': 1168, 'ROS10298957': 1169, 'MAP50112274': 1170, 'ROS13755811': 1171, 'ROS21000928': 1172, 'MAP30663686': 1173, 'MAP50406633': 1174, 'MAP18301541': 1175, 'MAP50107321': 1176, 'ROS21411585': 1177, 'ROS20170179': 1178, 'ROS20936511': 1179, 'MAP00617643': 1180, 'MAP02188675': 1181, 'ROS18584660': 1182, 'MAP24747976': 1183, 'ROS20254614': 1184, 'ROS21401713': 1185, 'MAP05001777': 1186, 'MAP39721045': 1187, 'MAP02657173': 1188, 'MAP00696418': 1189, 'ROS21274316': 1190, 'MAP02525608': 1191, 'MAP33707641': 1192, 'MAP84483565': 1193, 'MAP04330337': 1194, 'MAP31726180': 1195, 'MAP26938063': 1196, 'ROS21401001': 1197, 'ROS20153984': 1198, 'MAP03971682': 1199, 'ROS20524813': 1200, 'ROS37866777': 1201, 'ROS20455807': 1202, 'ROS20223617': 1203, 'MAP03335936': 1204, 'ROS20472416': 1205, 'MAP06272612': 1206, 'ROS15897112': 1207, 'ROS20433379': 1208, 'MAP90100426': 1209, 'ROS79626445': 1210, 'MAP69187342': 1211, 'ROS20160019': 1212, 'MAP39136480': 1213, 'ROS20293913': 1214, 'ROS20486354': 1215, 'MAP62301938': 1216, 'MAP64635241': 1217, 'ROS21409243': 1218, 'ROS20254588': 1219, 'MAP03430444': 1220, 'MAP72205714': 1221, 'ROS15846869': 1222, 'ROS20504017': 1223, 'ROS20447343': 1224, 'MAP05331374': 1225, 'ROS30544882': 1226, 'ROS20212416': 1227, 'MAP90133029': 1228, 'MAP05375330': 1229, 'MAP04127190': 1230, 'MAP19721447': 1231, 'MAP24951704': 1232, 'ROS20764143': 1233, 'MAP76755449': 1234, 'MAP92023910': 1235, 'MAP68419063': 1236, 'MAP35072859': 1237, 'MAP50100194': 1238, 'MAP73943687': 1239, 'MAP73976154': 1240, 'MAP75169847': 1241, 'ROS21188403': 1242, 'MAP46358797': 1243, 'ROS20537160': 1244, 'ROS21411611': 1245, 'ROS21409081': 1246, 'ROS21274154': 1247, 'MAP44041093': 1248, 'MAP04493359': 1249, 'MAP05522533': 1250, 'MAP23825005': 1251, 'MAP24680888': 1252, 'MAP25837301': 1253, 'MAP25939172': 1254, 'MAP26631069': 1255, 'MAP26637867': 1256, 'MAP72717425': 1257, 'MAP94734696': 1258, 'MAP58458351': 1259, 'MAP63188799': 1260, 'MAP70636113': 1261, 'MAP63357970': 1262, 'MAP68217991': 1263, 'MAP95600557': 1264, 'ROS20968785': 1265, 'MAP85328131': 1266, 'MAP33062311': 1267, 'MAP33321607': 1268, 'MAP33400773': 1269, 'ROS21163364': 1270, 'MAP34524842': 1271, 'MAP35705009': 1272, 'MAP39539033': 1273, 'MAP87555330': 1274, 'MAP43119512': 1275, 'MAP43536587': 1276, 'MAP44952317': 1277, 'MAP46246604': 1278, 'MAP63986591': 1279, 'MAP65685223': 1280, 'MAP66069855': 1281, 'ROS20583594': 1282, 'MAP72841884': 1283, 'MAP84732827': 1284, 'MAP85036828': 1285, 'MAP85980779': 1286, 'MAP66233829': 1287, 'ROS20594407': 1288, 'ROS21411747': 1289, 'ROS20197652': 1290, 'ROS21274442': 1291, 'ROS20522343': 1292, 'ROS20503840': 1293, 'ROS20109020': 1294, 'ROS20266406': 1295, 'ROS37874781': 1296, 'MAP65933570': 1297, 'MAP13485298': 1298, 'ROS20120255': 1299, 'MAP32959281': 1300, 'MAP33084399': 1301, 'MAP40530379': 1302, 'ROS20501668': 1303, 'ROS20534423': 1304, 'MAP47955338': 1305, 'MAP48192539': 1306, 'MAP49452862': 1307, 'MAP50102538': 1308, 'MAP50106442': 1309, 'MAP50108462': 1310, 'MAP50408527': 1311, 'MAP56214739': 1312, 'MAP56629174': 1313, 'ROS34779151': 1314, 'MAP75833578': 1315, 'MAP76137867': 1316, 'MAP78331049': 1317, 'MAP84248280': 1318, 'MAP86767530': 1319, 'MAP86947750': 1320, 'MAP40600002': 1321, 'MAP40880222': 1322, 'ROS20544321': 1323, 'ROS20257937': 1324, 'MAP86177506': 1325, 'MAP01797756': 1326, 'MAP03227207': 1327, 'MAP06769734': 1328, 'MAP06804844': 1329, 'MAP07265221': 1330, 'MAP08390339': 1331, 'MAP08604620': 1332, 'MAP13666711': 1333, 'MAP16446147': 1334, 'MAP18740882': 1335, 'MAP18942080': 1336, 'MAP26929170': 1337, 'MAP27242962': 1338, 'MAP28875005': 1339, 'MAP29067201': 1340, 'MAP29629849': 1341, 'ROS20901448': 1342, 'MAP79411152': 1343, 'MAP78410089': 1344, 'MAP80930779': 1345, 'ROS21000342': 1346, 'ROS20261613': 1347, 'MAP77631046': 1348, 'MAP77273914': 1349, 'MAP91525115': 1350, 'MAP81504393': 1351, 'MAP60725338': 1352, 'MAP99693576': 1353, 'ROS20235373': 1354, 'MAP71592202': 1355, 'MAP70153803': 1356, 'MAP65952361': 1357, 'MAP71537723': 1358, 'MAP98388248': 1359, 'MAP92427316': 1360, 'MAP60248664': 1361, 'ROS20279627': 1362, 'MAP94974890': 1363, 'MAP89811231': 1364, 'MAP93493992': 1365, 'MAP75675336': 1366, 'MAP63698192': 1367, 'ROS20674902': 1368, 'MAP84945776': 1369, 'MAP65736039': 1370, 'MAP77868579': 1371, 'MAP95723689': 1372, 'MAP49590907': 1373, 'MAP50100356': 1374, 'MAP50500000': 1375, 'MAP50500424': 1376, 'MAP50502606': 1377, 'MAP52322864': 1378, 'ROS20752487': 1379, 'MAP54571041': 1380, 'ROS20962563': 1381, 'MAP63649283': 1382, 'MAP41265021': 1383, 'MAP77689940': 1384, 'MAP53356828': 1385, 'MAP73177635': 1386, 'ROS21001357': 1387, 'MAP34108468': 1388, 'MAP50400259': 1389, 'MAP02136155': 1390, 'MAP46808242': 1391, 'MAP54519335': 1392, 'MAP63874408': 1393, 'MAP47520485': 1394, 'MAP50103679': 1395, 'MAP50304024': 1396, 'MAP50406769': 1397, 'MAP52018573': 1398, 'MAP52940446': 1399, 'MAP65214844': 1400, 'MAP04981936': 1401, 'MAP14452889': 1402, 'MAP88038410': 1403, 'ROS20283691': 1404, 'MAP24644776': 1405, 'MAP67725478': 1406, 'MAP51864085': 1407, 'MAP58171603': 1408, 'MAP50400411': 1409, 'MAP50502282': 1410, 'MAP50402279': 1411, 'ROS20110807': 1412, 'MAP49643021': 1413, 'MAP29278716': 1414, 'MAP52298793': 1415, 'MAP36379885': 1416, 'MAP40510997': 1417, 'MAP35170130': 1418, 'MAP39236493': 1419, 'MAP37997363': 1420, 'MAP50302130': 1421, 'ROS21411873': 1422, 'ROS10539331': 1423, 'MAP50106280': 1424, 'MAP50402855': 1425, 'ROS10218177': 1426, 'ROS10866165': 1427, 'MAP50108048': 1428, 'MAP50408077': 1429, 'ROS10415168': 1430, 'ROS10538326': 1431, 'MAP50106866': 1432, 'ROS10216869': 1433, 'MAP82478522': 1434, 'MAP48748345': 1435, 'MAP52731249': 1436, 'ROS11205014': 1437, 'MAP53422197': 1438, 'ROS20205705': 1439, 'ROS20153010': 1440, 'MAP15690969': 1441, 'MAP58331153': 1442, 'MAP50403158': 1443, 'ROS20218890': 1444, 'MAP31504924': 1445, 'MAP50408239': 1446, 'MAP12788806': 1447, 'MAP31728786': 1448, 'MAP37914710': 1449, 'MAP57311163': 1450, 'ROS10406301': 1451, 'MAP50300796': 1452, 'ROS10315029': 1453, 'ROS10551157': 1454, 'ROS10916128': 1455, 'MAP50300660': 1456, 'MAP50303307': 1457, 'ROS21412040': 1458, 'MAP50301701': 1459, 'MAP51903261': 1460, 'MAP55392868': 1461, 'MAP58143217': 1462, 'ROS20968361': 1463, 'MAP98096223': 1464, 'MAP58203880': 1465, 'MAP81455345': 1466, 'ROS97411796': 1467, 'ROS20695263': 1468, 'MAP77790442': 1469, 'MAP78005770': 1470, 'MAP79462793': 1471, 'MAP44019405': 1472, 'MAP80204332': 1473, 'MAP81403249': 1474, 'MAP02673641': 1475, 'ROS43596435': 1476, 'MAP91949579': 1477, 'MAP81783210': 1478, 'MAP86127970': 1479, 'MAP82455953': 1480, 'MAP82617423': 1481, 'MAP85584353': 1482, 'MAP86274610': 1483, 'MAP50402143': 1484, 'MAP50304998': 1485, 'MAP10093562': 1486, 'ROS20207013': 1487, 'MAP52957571': 1488, 'MAP40385646': 1489, 'MAP55025694': 1490, 'MAP50401390': 1491, 'MAP23004922': 1492, 'MAP35592862': 1493, 'MAP50199993': 1494, 'MAP08891975': 1495, 'ROS11198390': 1496, 'ROS11183735': 1497, 'ROS11240247': 1498, 'ROS10577337': 1499, 'ROS10394182': 1500, 'MAP50105301': 1501, 'ROS10473384': 1502, 'MAP50404037': 1503, 'ROS11045988': 1504, 'ROS10248033': 1505, 'MAP50407486': 1506, 'MAP14726154': 1507, 'MAP19460005': 1508, 'MAP19785885': 1509, 'MAP24219409': 1510, 'MAP26671823': 1511, 'MAP34326260': 1512, 'MAP35299510': 1513, 'MAP65384603': 1514, 'MAP75990666': 1515, 'MAP92928376': 1516, 'MAP44299049': 1517, 'MAP44951690': 1518, 'ROS20938081': 1519, 'MAP50107619': 1520, 'MAP71558084': 1521, 'MAP50302842': 1522, 'MAP75262623': 1523, 'ROS48319614': 1524, 'MAP50402981': 1525, 'MAP80368783': 1526, 'MAP80790863': 1527, 'MAP51791453': 1528, 'MAP53727207': 1529, 'MAP57442623': 1530, 'MAP38606123': 1531, 'MAP80487613': 1532, 'MAP58531593': 1533, 'MAP33332646': 1534, 'MAP62574447': 1535, 'ROS20928345': 1536, 'MAP65309742': 1537, 'MAP65925304': 1538, 'MAP65990079': 1539, 'MAP66394957': 1540, 'MAP67429065': 1541, 'ROS21193770': 1542, 'ROS32666549': 1543, 'ROS10484873': 1544, 'ROS20906231': 1545, 'MAP54324848': 1546, 'MAP53355949': 1547, 'MAP37617978': 1548, 'MAP00285563': 1549, 'MAP45514725': 1550, 'MAP13464351': 1551, 'MAP13747771': 1552, 'MAP05218242': 1553, 'MAP21362537': 1554, 'MAP54572768': 1555, 'MAP24073245': 1556, 'MAP45122535': 1557, 'MAP32383679': 1558, 'ROS10248169': 1559, 'MAP13447030': 1560, 'MAP33501827': 1561, 'MAP24141372': 1562, 'MAP18611216': 1563, 'MAP50101235': 1564, 'MAP50410445': 1565, 'MAP37759927': 1566, 'MAP50400961': 1567, 'MAP27083391': 1568, 'MAP45627085': 1569, 'MAP23993132': 1570, 'MAP50108912': 1571, 'MAP04767750': 1572, 'MAP50409956': 1573, 'MAP18920002': 1574, 'MAP08682642': 1575, 'MAP17705889': 1576, 'ROS10228075': 1577, 'MAP58014515': 1578, 'ROS11120276': 1579, 'MAP50303019': 1580, 'MAP50406183': 1581, 'ROS11322762': 1582, 'MAP50404451': 1583, 'MAP50401426': 1584, 'ROS10580684': 1585, 'ROS11082005': 1586, 'ROS10263208': 1587, 'MAP50408103': 1588, 'MAP50300534': 1589, 'ROS10479506': 1590, 'MAP50109053': 1591, 'MAP50402305': 1592, 'ROS11104122': 1593, 'MAP50104396': 1594, 'ROS10204365': 1595, 'MAP50112436': 1596, 'ROS21412176': 1597, 'ROS11160328': 1598, 'MAP50108336': 1599, 'ROS10430919': 1600, 'MAP44671043': 1601, 'ROS20124619': 1602, 'MAP08536593': 1603, 'MAP50105013': 1604, 'MAP05632732': 1605, 'MAP33006377': 1606, 'MAP50111971': 1607, 'ROS15101241': 1608, 'MAP35458631': 1609, 'MAP35110444': 1610, 'MAP39405962': 1611, 'MAP50401002': 1612, 'ROS10284569': 1613, 'MAP50403284': 1614, 'MAP50405042': 1615, 'MAP50403996': 1616, 'MAP50402693': 1617, 'ROS11145865': 1618, 'MAP50402729': 1619, 'ROS11349821': 1620, 'ROS11099004': 1621, 'ROS10577887': 1622, 'MAP14184286': 1623, 'ROS11119101': 1624, 'MAP17345449': 1625, 'MAP08337320': 1626, 'ROS21411909': 1627, 'ROS11039282': 1628, 'MAP07311370': 1629, 'MAP76431098': 1630, 'MAP29821047': 1631, 'MAP50302392': 1632, 'MAP73763467': 1633, 'MAP48697089': 1634, 'MAP92371267': 1635, 'MAP01266704': 1636, 'MAP50109639': 1637, 'ROS21274730': 1638, 'MAP50408491': 1639, 'MAP42063693': 1640, 'MAP50405330': 1641, 'ROS20254452': 1642, 'ROS20701008': 1643}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_25612/1663048613.py:4: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only.\n",
      "  usable_samples_ROSMAP = pd.read_csv(f'./PRSice_output_ROSMAP_2/{GWAS_ID}/{GWAS_ID}.best', ' ')['IID']\n"
     ]
    }
   ],
   "source": [
    "\n",
    "import json\n",
    "import pandas as pd\n",
    "GWAS_ID = 'ukb-b-13806'\n",
    "usable_samples_ROSMAP = pd.read_csv(f'./PRSice_output_ROSMAP_2/{GWAS_ID}/{GWAS_ID}.best', ' ')['IID']\n",
    "# print(np.array(usable_samples_ROSMAP))\n",
    "# \n",
    "# above line may need to be commented out\n",
    "usable_samples_ROSMAP_for_json=usable_samples_ROSMAP.to_numpy().tolist()\n",
    "# print(usable_samples_ROSMAP)\n",
    "usable_samples_ROSMAP_for_json={c:int(i) for i,c in enumerate(usable_samples_ROSMAP_for_json)}\n",
    "print(usable_samples_ROSMAP_for_json)\n",
    "json.dump(usable_samples_ROSMAP_for_json, open('./usable_samples_ROSMAP.json', 'w'))\n",
    "usable_samples_ROSMAP=np.array(usable_samples_ROSMAP)\n",
    "\n",
    "# print(type(usable_samples_ADNI), len(usable_samples_ADNI))\n",
    "# usable_samples_ADNI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ieu-b-109  met-d-Total_C  ukb-b-13806  ukb-b-17627  ukb-b-3957\tukb-b-770\r\n",
      "ieu-b-110  ukb-a-257\t  ukb-b-14057  ukb-b-18275  ukb-b-4424\tukb-b-8476\r\n",
      "ieu-b-111  ukb-b-11495\t  ukb-b-14180  ukb-b-19732  ukb-b-5779\tukb-d-20405_0\r\n",
      "ieu-b-25   ukb-b-12064\t  ukb-b-14699  ukb-b-19953  ukb-b-6134\tukb-d-20405_1\r\n",
      "ieu-b-38   ukb-b-12417\t  ukb-b-15541  ukb-b-20289  ukb-b-6324\tukb-d-20405_2\r\n",
      "ieu-b-39   ukb-b-12493\t  ukb-b-17006  ukb-b-2209   ukb-b-6358\r\n",
      "lib\t   ukb-b-12963\t  ukb-b-17243  ukb-b-323    ukb-b-7663\r\n"
     ]
    }
   ],
   "source": [
    "!ls PRSice_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1644, 52) 1644\n",
      "['ROS20275399' 'ROS10442701' 'ROS20152393' ... 'MAP50405330' 'ROS20254452'\n",
      " 'ROS20701008']\n"
     ]
    }
   ],
   "source": [
    "# import json\n",
    "# import numpy as np\n",
    "# # PRS_feature_matrix = np.load('./PRS_feature_matrix.npy')\n",
    "# PRS_feature_matrix = np.load('PRS_feature_matrix.npy')\n",
    "# # usable_samples_ADNI = json.load(open('./usable_samples_ADNI.json'))\n",
    "# print(PRS_feature_matrix.shape, usable_samples_ROSMAP.__len__())\n",
    "# print(usable_samples_ROSMAP)\n",
    "# # usable_samples_ADNI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# prsice result used in our model\n",
    "usable_samples_adni-> from .best of prsice corresponding to larger dataset we get the IIDs ( ie the individuals that we considered for our studies )\n",
    "now , larger data set is actually a part of ADNIMERGE dataset . So, here we have all the considered patients, but some of the data that are in ADNIMERGE are not present in larger dataset. That leads to the check of \"sample in usable_samples_adni\". \n",
    "The reason behind this is related to the generation of \"larger dataset\" generation from \"ADNI\" dataset.\n",
    "Here our main task is to find out those samples that have corresponding .best file entry, ie included in larger dataset."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final Samples Dementia length :  655\n",
      "Final Samples Non-Dementia length :  987\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(655, 987, 1642)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "Threshold = 0\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_AD_Dementia = set()\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "#         print(sample)\n",
    "#     if True:\n",
    "        \n",
    "        \n",
    "        Final_Samples_AD_Dementia = Final_Samples_AD_Dementia|{sample}\n",
    "#         Final_Samples_AD_Dementia = Final_Samples_AD_Dementia.union({sample})\n",
    "\n",
    "            \n",
    "                \n",
    "\n",
    "Final_Samples_Non_AD_Dementia = set()\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP:\n",
    "        Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final Samples Dementia length : \", len(Final_Samples_AD_Dementia) )\n",
    "print(\"Final Samples Non-Dementia length : \", len(Final_Samples_Non_AD_Dementia) )\n",
    "Final_Samples_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia]\n",
    "Final_Samples = Final_Samples_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "\n",
    "# Final_Samples = Final_Samples_AD_Dementia + Final_Samples_AD_Dementia\n",
    "# json.dump(Final_Samples, open(inputFolder+'Final_Samples.json', 'w'))\n",
    "json.dump(Final_Samples, open('Final_Samples_nores.json', 'w'))\n",
    "# json.dump(Final_Samples_AD_Dementia_with_label, open('Final_Samples_Dementia.json', 'w'))\n",
    "# json.dump(Final_Samples_Non_AD_Dementia_with_label, open('Final_Samples_NonDementia.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_with_label.__len__(), Final_Samples.__len__()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# AD / Non-AD generation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final Samples Dementia length :  655\n",
      "Final Samples Non-Dementia length :  858\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(655, 858, 1513)"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "Threshold = 2\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_AD_Dementia = set()\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "#         print(sample)\n",
    "#     if True:\n",
    "        \n",
    "        \n",
    "        Final_Samples_AD_Dementia = Final_Samples_AD_Dementia|{sample}\n",
    "#         Final_Samples_AD_Dementia = Final_Samples_AD_Dementia.union({sample})\n",
    "\n",
    "            \n",
    "                \n",
    "\n",
    "Final_Samples_Non_AD_Dementia = set()\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP:\n",
    "        Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final Samples Dementia length : \", len(Final_Samples_AD_Dementia) )\n",
    "print(\"Final Samples Non-Dementia length : \", len(Final_Samples_Non_AD_Dementia) )\n",
    "Final_Samples_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia]\n",
    "Final_Samples = Final_Samples_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "\n",
    "# Final_Samples = Final_Samples_AD_Dementia + Final_Samples_AD_Dementia\n",
    "# json.dump(Final_Samples, open(inputFolder+'Final_Samples.json', 'w'))\n",
    "json.dump(Final_Samples, open('Final_Samples_2yrs.json', 'w'))\n",
    "# json.dump(Final_Samples_AD_Dementia_with_label, open('Final_Samples_Dementia.json', 'w'))\n",
    "# json.dump(Final_Samples_Non_AD_Dementia_with_label, open('Final_Samples_NonDementia.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_with_label.__len__(), Final_Samples.__len__()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final Samples Dementia length :  655\n",
      "Final Samples Non-Dementia length :  739\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(655, 739, 1394)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# samples with no restriction\n",
    "# THRESHOLD_MONTH = 12*0\n",
    "Threshold = 4\n",
    "# long_rosmap = pd.read_csv('long_rosmap.csv')\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "# print(long_rosmap[\"projid\"].astype(str))\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_AD_Dementia = set()\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "#         print(sample)\n",
    "#     if True:\n",
    "        \n",
    "        \n",
    "        Final_Samples_AD_Dementia = Final_Samples_AD_Dementia|{sample}\n",
    "#         Final_Samples_AD_Dementia = Final_Samples_AD_Dementia.union({sample})\n",
    "\n",
    "            \n",
    "                \n",
    "\n",
    "Final_Samples_Non_AD_Dementia = set()\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP:\n",
    "        Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final Samples Dementia length : \", len(Final_Samples_AD_Dementia) )\n",
    "print(\"Final Samples Non-Dementia length : \", len(Final_Samples_Non_AD_Dementia) )\n",
    "Final_Samples_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia]\n",
    "Final_Samples = Final_Samples_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "\n",
    "# Final_Samples = Final_Samples_AD_Dementia + Final_Samples_AD_Dementia\n",
    "# json.dump(Final_Samples, open(inputFolder+'Final_Samples.json', 'w'))\n",
    "json.dump(Final_Samples, open('Final_Samples_4yrs.json', 'w'))\n",
    "# json.dump(Final_Samples_AD_Dementia_with_label, open('Final_Samples_Dementia.json', 'w'))\n",
    "# json.dump(Final_Samples_Non_AD_Dementia_with_label, open('Final_Samples_NonDementia.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_with_label.__len__(), Final_Samples.__len__()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final Samples Dementia length :  655\n",
      "Final Samples Non-Dementia length :  631\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(655, 631, 1286)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "Threshold = 6\n",
    "# long_rosmap = pd.read_csv('long_rosmap.csv')\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "# print(long_rosmap[\"projid\"].astype(str))\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_AD_Dementia = set()\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "#         print(sample)\n",
    "#     if True:\n",
    "        \n",
    "        \n",
    "        Final_Samples_AD_Dementia = Final_Samples_AD_Dementia|{sample}\n",
    "#         Final_Samples_AD_Dementia = Final_Samples_AD_Dementia.union({sample})\n",
    "\n",
    "            \n",
    "                \n",
    "\n",
    "Final_Samples_Non_AD_Dementia = set()\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP:\n",
    "        Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final Samples Dementia length : \", len(Final_Samples_AD_Dementia) )\n",
    "print(\"Final Samples Non-Dementia length : \", len(Final_Samples_Non_AD_Dementia) )\n",
    "Final_Samples_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia]\n",
    "Final_Samples = Final_Samples_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "\n",
    "# Final_Samples = Final_Samples_AD_Dementia + Final_Samples_AD_Dementia\n",
    "# json.dump(Final_Samples, open(inputFolder+'Final_Samples.json', 'w'))\n",
    "json.dump(Final_Samples, open('Final_Samples_6yrs.json', 'w'))\n",
    "json.dump(Final_Samples_AD_Dementia_with_label, open('Final_Samples_Dementia.json', 'w'))\n",
    "json.dump(Final_Samples_Non_AD_Dementia_with_label, open('Final_Samples_NonDementia.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_with_label.__len__(), Final_Samples.__len__()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final Samples Dementia length :  655\n",
      "Final Samples Non-Dementia length :  575\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(655, 575, 1230)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "Threshold = 7\n",
    "# long_rosmap = pd.read_csv('long_rosmap.csv')\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "# print(long_rosmap[\"projid\"].astype(str))\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_AD_Dementia = set()\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "#         print(sample)\n",
    "#     if True:\n",
    "        \n",
    "        \n",
    "        Final_Samples_AD_Dementia = Final_Samples_AD_Dementia|{sample}\n",
    "#         Final_Samples_AD_Dementia = Final_Samples_AD_Dementia.union({sample})\n",
    "\n",
    "            \n",
    "                \n",
    "\n",
    "Final_Samples_Non_AD_Dementia = set()\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP:\n",
    "        Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final Samples Dementia length : \", len(Final_Samples_AD_Dementia) )\n",
    "print(\"Final Samples Non-Dementia length : \", len(Final_Samples_Non_AD_Dementia) )\n",
    "Final_Samples_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia]\n",
    "Final_Samples = Final_Samples_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "\n",
    "# Final_Samples = Final_Samples_AD_Dementia + Final_Samples_AD_Dementia\n",
    "# json.dump(Final_Samples, open(inputFolder+'Final_Samples.json', 'w'))\n",
    "json.dump(Final_Samples, open('Final_Samples_7yrs.json', 'w'))\n",
    "json.dump(Final_Samples_AD_Dementia_with_label, open('Final_Samples_Dementia.json', 'w'))\n",
    "json.dump(Final_Samples_Non_AD_Dementia_with_label, open('Final_Samples_NonDementia.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_with_label.__len__(), Final_Samples.__len__()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final Samples Dementia length :  655\n",
      "Final Samples Non-Dementia length :  537\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(655, 537, 1192)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "Threshold = 8\n",
    "# long_rosmap = pd.read_csv('long_rosmap.csv')\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "# print(long_rosmap[\"projid\"].astype(str))\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_AD_Dementia = set()\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "#         print(sample)\n",
    "#     if True:\n",
    "        \n",
    "        \n",
    "        Final_Samples_AD_Dementia = Final_Samples_AD_Dementia|{sample}\n",
    "#         Final_Samples_AD_Dementia = Final_Samples_AD_Dementia.union({sample})\n",
    "\n",
    "            \n",
    "                \n",
    "\n",
    "Final_Samples_Non_AD_Dementia = set()\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP:\n",
    "        Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final Samples Dementia length : \", len(Final_Samples_AD_Dementia) )\n",
    "print(\"Final Samples Non-Dementia length : \", len(Final_Samples_Non_AD_Dementia) )\n",
    "Final_Samples_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia]\n",
    "Final_Samples = Final_Samples_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "\n",
    "# Final_Samples = Final_Samples_AD_Dementia + Final_Samples_AD_Dementia\n",
    "# json.dump(Final_Samples, open(inputFolder+'Final_Samples.json', 'w'))\n",
    "json.dump(Final_Samples, open('Final_Samples_8yrs.json', 'w'))\n",
    "# json.dump(Final_Samples_AD_Dementia_with_label, open('Final_Samples_Dementia.json', 'w'))\n",
    "# json.dump(Final_Samples_Non_AD_Dementia_with_label, open('Final_Samples_NonDementia.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_with_label.__len__(), Final_Samples.__len__()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Progression from non-dementia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final_Samples_Non_AD_Dementia_to_AD_Dementia length :  553\n",
      "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length :  739\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(553, 739, 1292)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 4\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia = set()\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = set()\n",
    "\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:  \n",
    "        Final_Samples_Non_AD_Dementia_to_AD_Dementia = Final_Samples_Non_AD_Dementia_to_AD_Dementia|{sample}\n",
    "\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:\n",
    "        Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia|{sample}\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_AD_Dementia) )\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia) )\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_Non_AD_Dementia_to_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia]\n",
    "Final_Samples_progress_from_non_dem = Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "json.dump(Final_Samples_progress_from_non_dem, open('Final_Samples_4yrs_transition_from_Non_dementia_rosmap.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label.__len__(), Final_Samples_progress_from_non_dem.__len__()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final_Samples_Non_AD_Dementia_to_AD_Dementia length :  553\n",
      "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length :  631\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(553, 631, 1184)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 6\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia = set()\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = set()\n",
    "\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:  \n",
    "        Final_Samples_Non_AD_Dementia_to_AD_Dementia = Final_Samples_Non_AD_Dementia_to_AD_Dementia|{sample}\n",
    "\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:\n",
    "        Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia|{sample}\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_AD_Dementia) )\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia) )\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_Non_AD_Dementia_to_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia]\n",
    "Final_Samples_progress_from_non_dem = Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "json.dump(Final_Samples_progress_from_non_dem, open('Final_Samples_6yrs_transition_from_Non_dementia_rosmap.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label.__len__(), Final_Samples_progress_from_non_dem.__len__()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final_Samples_Non_AD_Dementia_to_AD_Dementia length :  553\n",
      "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length :  537\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(553, 537, 1090)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 8\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia = set()\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = set()\n",
    "\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:  \n",
    "        Final_Samples_Non_AD_Dementia_to_AD_Dementia = Final_Samples_Non_AD_Dementia_to_AD_Dementia|{sample}\n",
    "\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:\n",
    "        Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia|{sample}\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_AD_Dementia) )\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia) )\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_Non_AD_Dementia_to_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia]\n",
    "Final_Samples_progress_from_non_dem = Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "json.dump(Final_Samples_progress_from_non_dem, open('Final_Samples_8yrs_transition_from_Non_dementia_rosmap.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label.__len__(), Final_Samples_progress_from_non_dem.__len__()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final_Samples_Non_AD_Dementia_to_AD_Dementia length :  553\n",
      "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length :  446\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(553, 446, 999)"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 10\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia = set()\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = set()\n",
    "\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:  \n",
    "        Final_Samples_Non_AD_Dementia_to_AD_Dementia = Final_Samples_Non_AD_Dementia_to_AD_Dementia|{sample}\n",
    "\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:\n",
    "        Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia|{sample}\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_AD_Dementia) )\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia) )\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_Non_AD_Dementia_to_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia]\n",
    "Final_Samples_progress_from_non_dem = Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "json.dump(Final_Samples_progress_from_non_dem, open('Final_Samples_10yrs_transition_from_Non_dementia_rosmap.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label.__len__(), Final_Samples_progress_from_non_dem.__len__()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Dementia shape :  (3633, 3)\n",
      "Samples Dementia length :  1210\n",
      "Samples NonDementia length :  2531\n",
      "Final_Samples_Non_AD_Dementia_to_AD_Dementia length :  553\n",
      "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length :  363\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(553, 363, 916)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 12\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "# Month -> fuyear\n",
    "# dx -> dcfdx\n",
    "# projid\tfu_year\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "# dcfdx numbers\n",
    "# dcfdx can be numbers. 1 to 6 . 4 or 5 means AD\n",
    "\n",
    "AD = rosmap_temp[rosmap_temp['dcfdx']==4]\n",
    "ADPLUS=rosmap_temp[rosmap_temp['dcfdx']==5]\n",
    "AD_Dementia=pd.concat([AD,ADPLUS])\n",
    "OtherDementia=rosmap_temp[rosmap_temp['dcfdx']==6]\n",
    "print(\"Dementia shape : \", AD_Dementia.shape)\n",
    "# print(Dementia)\n",
    "Samples_AD_Dementia = set(AD_Dementia['PTID'].unique())\n",
    "Samples_otherDementia = set(OtherDementia['PTID'].unique())\n",
    "Samples_Non_AD_Dementia = set(rosmap_temp['PTID'].unique()) - Samples_AD_Dementia\n",
    "print(\"Samples Dementia length : \", len(Samples_AD_Dementia) )\n",
    "print(\"Samples NonDementia length : \", len( Samples_Non_AD_Dementia ) )\n",
    "# print(\"AD dementia samples\",Samples_AD_Dementia)\n",
    "# let's assume that we have a list of usable samples from ROSMAP\n",
    "# # Samples_Non_AD_Dementia\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia = set()\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = set()\n",
    "\n",
    "for sample in Samples_AD_Dementia:\n",
    "   \n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "#     print(sample)\n",
    "\n",
    "    if sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:  \n",
    "        Final_Samples_Non_AD_Dementia_to_AD_Dementia = Final_Samples_Non_AD_Dementia_to_AD_Dementia|{sample}\n",
    "\n",
    "for sample in Samples_Non_AD_Dementia:\n",
    "    dx_bl= rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[0]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "#     print(last_year_of_dx)\n",
    "    if last_year_of_dx >= Threshold and sample in usable_samples_ROSMAP and dx_bl != 4 and dx_bl != 5:\n",
    "        Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia = Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia|{sample}\n",
    "#         Final_Samples_Non_AD_Dementia = Final_Samples_Non_AD_Dementia | {sample} # union symbol in python sets\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_AD_Dementia) )\n",
    "print(\"Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia length : \", len(Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia) )\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label = [[x, 1] for x in Final_Samples_Non_AD_Dementia_to_AD_Dementia]\n",
    "Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label = [[x, 0] for x in Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia]\n",
    "Final_Samples_progress_from_non_dem = Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label + Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label\n",
    "\n",
    "\n",
    "json.dump(Final_Samples_progress_from_non_dem, open('Final_Samples_12yrs_transition_from_Non_dementia_rosmap.json', 'w'))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Final_Samples_Non_AD_Dementia_to_AD_Dementia_with_label.__len__(), Final_Samples_Non_AD_Dementia_to_Non_AD_Dementia_with_label.__len__(), Final_Samples_progress_from_non_dem.__len__()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Progression from CN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Baseline CN data size  3098\n",
      "samples progressed:  724\n",
      "samples not progressed:  490\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 4\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "all_samples = set(rosmap_temp['PTID'].unique())\n",
    "all_cn_samples= rosmap_temp[rosmap_temp['dcfdx']==1]\n",
    "all_cn_samples=set(all_cn_samples['PTID'].unique())\n",
    "print(\"Baseline CN data size \",all_cn_samples.__len__())\n",
    "\n",
    "samples_progressed= set()\n",
    "samples_not_progressed= set()\n",
    "for sample in all_cn_samples :\n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "        if (last_dx != 1):\n",
    "                samples_progressed = samples_progressed|{sample}\n",
    "        elif last_year_of_dx >= Threshold :\n",
    "                samples_not_progressed = samples_not_progressed|{sample}\n",
    "print(\"samples progressed: \", len(samples_progressed))\n",
    "print(\"samples not progressed: \", len(samples_not_progressed))\n",
    "samples_progressed = [[x, 1] for x in samples_progressed]\n",
    "samples_not_progressed = [[x, 0] for x in samples_not_progressed]\n",
    "final_samples=samples_progressed+samples_not_progressed\n",
    "json.dump(final_samples, open('final_samples_progression_from_cn_4yrs_rosmap.json', 'w'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Baseline CN data size  3098\n",
      "samples progressed:  724\n",
      "samples not progressed:  420\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 6\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "all_samples = set(rosmap_temp['PTID'].unique())\n",
    "all_cn_samples= rosmap_temp[rosmap_temp['dcfdx']==1]\n",
    "all_cn_samples=set(all_cn_samples['PTID'].unique())\n",
    "print(\"Baseline CN data size \",all_cn_samples.__len__())\n",
    "\n",
    "samples_progressed= set()\n",
    "samples_not_progressed= set()\n",
    "for sample in all_cn_samples :\n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "        if (last_dx != 1):\n",
    "                samples_progressed = samples_progressed|{sample}\n",
    "        elif last_year_of_dx >= Threshold :\n",
    "                samples_not_progressed = samples_not_progressed|{sample}\n",
    "print(\"samples progressed: \", len(samples_progressed))\n",
    "print(\"samples not progressed: \", len(samples_not_progressed))\n",
    "samples_progressed = [[x, 1] for x in samples_progressed]\n",
    "samples_not_progressed = [[x, 0] for x in samples_not_progressed]\n",
    "final_samples=samples_progressed+samples_not_progressed\n",
    "json.dump(final_samples, open('final_samples_progression_from_cn_6yrs_rosmap.json', 'w'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Baseline CN data size  3098\n",
      "samples progressed:  724\n",
      "samples not progressed:  365\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 8\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "all_samples = set(rosmap_temp['PTID'].unique())\n",
    "all_cn_samples= rosmap_temp[rosmap_temp['dcfdx']==1]\n",
    "all_cn_samples=set(all_cn_samples['PTID'].unique())\n",
    "print(\"Baseline CN data size \",all_cn_samples.__len__())\n",
    "\n",
    "samples_progressed= set()\n",
    "samples_not_progressed= set()\n",
    "for sample in all_cn_samples :\n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "        if (last_dx != 1):\n",
    "                samples_progressed = samples_progressed|{sample}\n",
    "        elif last_year_of_dx >= Threshold :\n",
    "                samples_not_progressed = samples_not_progressed|{sample}\n",
    "print(\"samples progressed: \", len(samples_progressed))\n",
    "print(\"samples not progressed: \", len(samples_not_progressed))\n",
    "samples_progressed = [[x, 1] for x in samples_progressed]\n",
    "samples_not_progressed = [[x, 0] for x in samples_not_progressed]\n",
    "final_samples=samples_progressed+samples_not_progressed\n",
    "json.dump(final_samples, open('final_samples_progression_from_cn_8yrs_rosmap.json', 'w'))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Progression from MCI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Baseline MCI data size  2181\n",
      "samples progressed:  474\n",
      "samples not progressed:  481\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 4\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "all_samples = set(rosmap_temp['PTID'].unique())\n",
    "all_mci_samples1=rosmap_temp[rosmap_temp['dcfdx']==2]\n",
    "all_mci_samples1 = set(all_mci_samples1['PTID'].unique())\n",
    "all_mci_samples2=rosmap_temp[rosmap_temp['dcfdx']==3]\n",
    "all_mci_samples2 = set(all_mci_samples2['PTID'].unique())\n",
    "all_mci_samples = all_mci_samples1 | all_mci_samples2\n",
    "print(\"Baseline MCI data size \",all_mci_samples.__len__())\n",
    "\n",
    "samples_progressed= set()\n",
    "samples_not_progressed= set()\n",
    "for sample in all_mci_samples :\n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "        if (last_dx > 3 and last_dx<6):\n",
    "                samples_progressed = samples_progressed|{sample}\n",
    "        elif last_year_of_dx >= Threshold :\n",
    "                samples_not_progressed = samples_not_progressed|{sample}\n",
    "print(\"samples progressed: \", len(samples_progressed))\n",
    "print(\"samples not progressed: \", len(samples_not_progressed))\n",
    "samples_progressed = [[x, 1] for x in samples_progressed]\n",
    "samples_not_progressed = [[x, 0] for x in samples_not_progressed]\n",
    "final_samples=samples_progressed+samples_not_progressed\n",
    "json.dump(final_samples, open('final_samples_progression_from_mci_4yrs_rosmap.json', 'w'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Baseline MCI data size  2181\n",
      "samples progressed:  474\n",
      "samples not progressed:  416\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 6\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "all_samples = set(rosmap_temp['PTID'].unique())\n",
    "all_mci_samples1=rosmap_temp[rosmap_temp['dcfdx']==2]\n",
    "all_mci_samples1 = set(all_mci_samples1['PTID'].unique())\n",
    "all_mci_samples2=rosmap_temp[rosmap_temp['dcfdx']==3]\n",
    "all_mci_samples2 = set(all_mci_samples2['PTID'].unique())\n",
    "all_mci_samples = all_mci_samples1 | all_mci_samples2\n",
    "print(\"Baseline MCI data size \",all_mci_samples.__len__())\n",
    "\n",
    "samples_progressed= set()\n",
    "samples_not_progressed= set()\n",
    "for sample in all_mci_samples :\n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "        if (last_dx > 3 and last_dx<6):\n",
    "                samples_progressed = samples_progressed|{sample}\n",
    "        elif last_year_of_dx >= Threshold :\n",
    "                samples_not_progressed = samples_not_progressed|{sample}\n",
    "print(\"samples progressed: \", len(samples_progressed))\n",
    "print(\"samples not progressed: \", len(samples_not_progressed))\n",
    "samples_progressed = [[x, 1] for x in samples_progressed]\n",
    "samples_not_progressed = [[x, 0] for x in samples_not_progressed]\n",
    "final_samples=samples_progressed+samples_not_progressed\n",
    "json.dump(final_samples, open('final_samples_progression_from_mci_6yrs_rosmap.json', 'w'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Baseline MCI data size  2181\n",
      "samples progressed:  474\n",
      "samples not progressed:  348\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 8\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "all_samples = set(rosmap_temp['PTID'].unique())\n",
    "all_mci_samples1=rosmap_temp[rosmap_temp['dcfdx']==2]\n",
    "all_mci_samples1 = set(all_mci_samples1['PTID'].unique())\n",
    "all_mci_samples2=rosmap_temp[rosmap_temp['dcfdx']==3]\n",
    "all_mci_samples2 = set(all_mci_samples2['PTID'].unique())\n",
    "all_mci_samples = all_mci_samples1 | all_mci_samples2\n",
    "print(\"Baseline MCI data size \",all_mci_samples.__len__())\n",
    "\n",
    "samples_progressed= set()\n",
    "samples_not_progressed= set()\n",
    "for sample in all_mci_samples :\n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "        if (last_dx > 3 and last_dx<6):\n",
    "                samples_progressed = samples_progressed|{sample}\n",
    "        elif last_year_of_dx >= Threshold :\n",
    "                samples_not_progressed = samples_not_progressed|{sample}\n",
    "print(\"samples progressed: \", len(samples_progressed))\n",
    "print(\"samples not progressed: \", len(samples_not_progressed))\n",
    "samples_progressed = [[x, 1] for x in samples_progressed]\n",
    "samples_not_progressed = [[x, 0] for x in samples_not_progressed]\n",
    "final_samples=samples_progressed+samples_not_progressed\n",
    "json.dump(final_samples, open('final_samples_progression_from_mci_8yrs_rosmap.json', 'w'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n",
      "/home/bayzid/anaconda3/envs/ad_venv_2/lib/python3.10/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default\n",
      "  warn(\"Workbook contains no default style, apply openpyxl's default\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31534, 26)\n",
      "Baseline MCI data size  2181\n",
      "samples progressed:  474\n",
      "samples not progressed:  281\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "Threshold = 10\n",
    "long_rosmap=pd.read_excel('dataset_1225_long.xlsx', dtype={\"projid\":\"string\"})\n",
    "basic_rosmap=pd.read_excel('dataset_1225_basic.xlsx')\n",
    "long_rosmap[\"PTID\"]=long_rosmap[\"study\"].str.replace(\" \",\"\")+long_rosmap[\"projid\"].astype(str)\n",
    "# print(long_rosmap[\"PTID\"])\n",
    "\n",
    "print(long_rosmap.shape)\n",
    "rosmap_temp = long_rosmap[['PTID', 'dcfdx', 'fu_year']].dropna().sort_values(by=['PTID', 'fu_year'])\n",
    "all_samples = set(rosmap_temp['PTID'].unique())\n",
    "all_mci_samples1=rosmap_temp[rosmap_temp['dcfdx']==2]\n",
    "all_mci_samples1 = set(all_mci_samples1['PTID'].unique())\n",
    "all_mci_samples2=rosmap_temp[rosmap_temp['dcfdx']==3]\n",
    "all_mci_samples2 = set(all_mci_samples2['PTID'].unique())\n",
    "all_mci_samples = all_mci_samples1 | all_mci_samples2\n",
    "print(\"Baseline MCI data size \",all_mci_samples.__len__())\n",
    "\n",
    "samples_progressed= set()\n",
    "samples_not_progressed= set()\n",
    "for sample in all_mci_samples :\n",
    "    last_dx = rosmap_temp[rosmap_temp['PTID']==sample]['dcfdx'].iloc[-1]\n",
    "    last_year_of_dx = rosmap_temp[rosmap_temp['PTID']==sample]['fu_year'].iloc[-1]\n",
    "\n",
    "    if sample in usable_samples_ROSMAP:\n",
    "        if (last_dx > 3 and last_dx<6):\n",
    "                samples_progressed = samples_progressed|{sample}\n",
    "        elif last_year_of_dx >= Threshold :\n",
    "                samples_not_progressed = samples_not_progressed|{sample}\n",
    "print(\"samples progressed: \", len(samples_progressed))\n",
    "print(\"samples not progressed: \", len(samples_not_progressed))\n",
    "samples_progressed = [[x, 1] for x in samples_progressed]\n",
    "samples_not_progressed = [[x, 0] for x in samples_not_progressed]\n",
    "final_samples=samples_progressed+samples_not_progressed\n",
    "json.dump(final_samples, open('final_samples_progression_from_mci_10yrs_rosmap.json', 'w'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ad_venv_2",
   "language": "python",
   "name": "ad_venv_2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "vscode": {
   "interpreter": {
    "hash": "916dbcbb3f70747c44a77c7bcd40155683ae19c65e1c03b4aa3499c5328201f1"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
